diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 9ba346961..d90895f2f 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -8,14 +8,6 @@ ENV['RACK_ENV'] = 'test'

 require 'bundler/setup'

-require 'simplecov'
-
-# Configure SimpleCov
-SimpleCov.start do
-  track_files 'lib/**/*.rb'
-  add_filter 'spec'
-end
-
 require 'open3'
 require 'thor'
 require 'fileutils'
@@ -37,9 +29,6 @@ if unit_tests
   Dir.glob(File.join('./lib/facter/resolvers', '/**/*/', '*.rb')).sort.each(&method(:require))
 end

-default_coverage = 90
-SimpleCov.minimum_coverage ENV['COVERAGE'] || default_coverage
-
 def colorize(str, color)
   "#{color}#{str}#{Facter::RESET}"
 end
