diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index e42c4fb9..1d4f4665 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -17,7 +17,7 @@ set(wrapper_SRCS
 set(_python_module_install_dir "avogadro")
 # SKBUILD is set for binary wheel
 if (NOT SKBUILD)
-  set(_python_module_install_dir "${Python_STDARCH}/avogadro")
+  set(_python_module_install_dir "${Python_SITEARCH}/avogadro")
 endif()
 
 set(CMAKE_MODULE_LINKER_FLAGS "")
