diff --git a/src/fileutils.cpp b/src/fileutils.cpp
index f9821bec2..7ba9ad05f 100644
--- a/src/fileutils.cpp
+++ b/src/fileutils.cpp
@@ -67,6 +67,7 @@
 #endif
 
 #include <nettle/sha3.h>
+#include <nettle/version.h>
 
 #include <sstream>
 #include <fstream>
@@ -796,7 +797,11 @@ sha3File(const std::filesystem::path& path)
     }
 
     unsigned char digest[SHA3_512_DIGEST_SIZE];
+#if NETTLE_VERSION_MAJOR > 3
+    sha3_512_digest(&ctx, digest);
+#else
     sha3_512_digest(&ctx, SHA3_512_DIGEST_SIZE, digest);
+#endif
 
     char hash[SHA3_512_DIGEST_SIZE * 2];
 
@@ -813,7 +818,11 @@ sha3sum(const uint8_t* data, size_t size)
     sha3_512_init(&ctx);
     sha3_512_update(&ctx, size, data);
     unsigned char digest[SHA3_512_DIGEST_SIZE];
+#if NETTLE_VERSION_MAJOR > 3
+    sha3_512_digest(&ctx, digest);
+#else
     sha3_512_digest(&ctx, SHA3_512_DIGEST_SIZE, digest);
+#endif
     return dht::toHex(digest, SHA3_512_DIGEST_SIZE);
 }
 

