diff -ru profanity-0.15.0.orig/src/pgp/gpg.c profanity-0.15.0/src/pgp/gpg.c
--- profanity-0.15.0.orig/src/pgp/gpg.c	2025-06-04 12:46:15.017497664 +0200
+++ profanity-0.15.0/src/pgp/gpg.c	2025-06-04 12:50:22.028087967 +0200
@@ -656,7 +656,7 @@
             error = gpgme_get_key(ctx, recipient->keyid, &key, 1);
 
             if (!error && key) {
-                const char* addr = gpgme_key_get_string_attr(key, GPGME_ATTR_EMAIL, NULL, 0);
+                const char* addr = key->uids->email;
                 if (addr) {
                     g_string_append(recipients_str, addr);
                 }
