diff --git a/pyproject.toml b/pyproject.toml
index 84f2370..9622f35 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,12 +32,10 @@ Documentation = "https://frankie567.github.io/httpx-ws/"
 Source = "https://github.com/frankie567/httpx-ws"
 
 [build-system]
-requires = ["hatchling", "hatch-regex-commit"]
+requires = ["hatchling"]
 build-backend = "hatchling.build"
 
 [tool.hatch.version]
-source = "regex_commit"
-commit_extra_args = ["-e"]
 path = "httpx_ws/__init__.py"
 
 [tool.hatch.build.targets.sdist]
