diff --git a/src/VBox/Devices/Graphics/DevVGA-SVGA.cpp b/src/VBox/Devices/Graphics/DevVGA-SVGA.cpp
--- a/src/VBox/Devices/Graphics/DevVGA-SVGA.cpp
+++ b/src/VBox/Devices/Graphics/DevVGA-SVGA.cpp
@@ -7082,7 +7082,7 @@
 
     if (!pThis->svga.fVMSVGA2dGBO)
     {
-        for (unsigned i = 0; i <= RT_ELEMENTS(pThis->svga.au32DevCaps); ++i)
+        for (unsigned i = 0; i < RT_ELEMENTS(pThis->svga.au32DevCaps); ++i)
         {
             uint32_t val = 0;
             int rc = vmsvga3dQueryCaps(pThisCC, (SVGA3dDevCapIndex)i, &val);
