 Modules/GNUInstallDirs.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Modules/GNUInstallDirs.cmake b/Modules/GNUInstallDirs.cmake
index ff9c11a9d7..e0372e673d 100644
--- a/Modules/GNUInstallDirs.cmake
+++ b/Modules/GNUInstallDirs.cmake
@@ -343,6 +343,8 @@ function(_GNUInstallDirs_get_system_type_for_install out_var)
       set(${out_var} "alpine")
     elseif (EXISTS "/etc/arch-release")
       set(${out_var} "arch linux")
+    elseif (EXISTS "/etc/artix-release")
+      set(${out_var} "artix linux")
     elseif (EXISTS "/etc/debian_version")
       set(${out_var} "debian")
     endif()
