diff -ru kquickimageeditor-0.6.2.orig/CMakeLists.txt kquickimageeditor-0.6.2/CMakeLists.txt
--- kquickimageeditor-0.6.2.orig/CMakeLists.txt	2026-06-19 22:42:35.619263525 +0200
+++ kquickimageeditor-0.6.2/CMakeLists.txt	2026-06-19 23:06:20.212668427 +0200
@@ -14,6 +14,7 @@
 
 include(FeatureSummary)
 include(CMakePackageConfigHelpers)
+include(CMakeDependentOption)
 
 include(KDEInstallDirs)
 include(KDECMakeSettings)
@@ -32,7 +33,7 @@
 find_package(Qt6 ${REQUIRED_QT_VERSION} COMPONENTS Core Qml Quick REQUIRED)
 find_package(KF6 ${REQUIRED_KF_VERSION} REQUIRED Config)
 
-option(WITH_OPENCV "Build with OpenCV stack blur (recommended)" UNIX)
+cmake_dependent_option(WITH_OPENCV "Build with OpenCV stack blur (recommended)" TRUE UNIX FALSE)
 if(WITH_OPENCV)
     find_package(OpenCV 4.7 COMPONENTS core imgproc REQUIRED)
 endif()
