diff --git i/testdata/testoutput8-16-2 w/testdata/testoutput8-16-2
index bcb9e17..8825886 100644
--- i/testdata/testoutput8-16-2
+++ w/testdata/testoutput8-16-2
@@ -10,7 +10,7 @@
 #pattern fullbincode,memory
 
 /((?i)b)/
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -22,7 +22,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /(?s)(.*X|^B)/
-Memory allocation - compiled block : 174
+Memory allocation - compiled block : 150
 Memory allocation - code portion   : 38
 ------------------------------------------------------------------
   0  16 Bra
@@ -38,7 +38,7 @@ Memory allocation - code portion   : 38
 ------------------------------------------------------------------
 
 /(?s:.*X|^B)/
-Memory allocation - compiled block : 172
+Memory allocation - compiled block : 148
 Memory allocation - code portion   : 36
 ------------------------------------------------------------------
   0  15 Bra
@@ -54,7 +54,7 @@ Memory allocation - code portion   : 36
 ------------------------------------------------------------------
 
 /^[[:alnum:]]/
-Memory allocation - compiled block : 182
+Memory allocation - compiled block : 158
 Memory allocation - code portion   : 46
 ------------------------------------------------------------------
   0  20 Bra
@@ -65,7 +65,7 @@ Memory allocation - code portion   : 46
 ------------------------------------------------------------------
 
 /#/Ix
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   2 Bra
@@ -78,7 +78,7 @@ Options: extended
 Subject length lower bound = 0
 
 /a#/Ix
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -92,7 +92,7 @@ First code unit = 'a'
 Subject length lower bound = 1
 
 /x?+/
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -102,7 +102,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /x++/
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -112,7 +112,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /x{1,3}+/
-Memory allocation - compiled block : 156
+Memory allocation - compiled block : 132
 Memory allocation - code portion   : 20
 ------------------------------------------------------------------
   0   7 Bra
@@ -123,7 +123,7 @@ Memory allocation - code portion   : 20
 ------------------------------------------------------------------
 
 /(x)*+/
-Memory allocation - compiled block : 162
+Memory allocation - compiled block : 138
 Memory allocation - code portion   : 26
 ------------------------------------------------------------------
   0  10 Bra
@@ -136,7 +136,7 @@ Memory allocation - code portion   : 26
 ------------------------------------------------------------------
 
 /^((a+)(?U)([ab]+)(?-U)([bc]+)(\w*))/
-Memory allocation - compiled block : 278
+Memory allocation - compiled block : 254
 Memory allocation - code portion   : 142
 ------------------------------------------------------------------
   0  68 Bra
@@ -160,7 +160,7 @@ Memory allocation - code portion   : 142
 ------------------------------------------------------------------
 
 "8J\$WE\<\.rX\+ix\[d1b\!H\#\?vV0vrK\:ZH1\=2M\>iV\;\?aPhFB\<\*vW\@QW\@sO9\}cfZA\-i\'w\%hKd6gt1UJP\,15_\#QY\$M\^Mss_U\/\]\&LK9\[5vQub\^w\[KDD\<EjmhUZ\?\.akp2dF\>qmj\;2\}YWFdYx\.Ap\]hjCPTP\(n28k\+3\;o\&WXqs\/gOXdr\$\:r\'do0\;b4c\(f_Gr\=\"\\4\)\[01T7ajQJvL\$W\~mL_sS\/4h\:x\*\[ZN\=KLs\&L5zX\/\/\>it\,o\:aU\(\;Z\>pW\&T7oP\'2K\^E\:x9\'c\[\%z\-\,64JQ5AeH_G\#KijUKghQw\^\\vea3a\?kka_G\$8\#\`\*kynsxzBLru\'\]k_\[7FrVx\}\^\=\$blx\>s\-N\%j\;D\*aZDnsw\:YKZ\%Q\.Kne9\#hP\?\+b3\(SOvL\,\^\;\&u5\@\?5C5Bhb\=m\-vEh_L15Jl\]U\)0RP6\{q\%L\^_z5E\'Dw6X\b"
-Memory allocation - compiled block : 1784
+Memory allocation - compiled block : 1760
 Memory allocation - code portion   : 1648
 ------------------------------------------------------------------
   0 821 Bra
@@ -171,7 +171,7 @@ Memory allocation - code portion   : 1648
 ------------------------------------------------------------------
 
 "\$\<\.X\+ix\[d1b\!H\#\?vV0vrK\:ZH1\=2M\>iV\;\?aPhFB\<\*vW\@QW\@sO9\}cfZA\-i\'w\%hKd6gt1UJP\,15_\#QY\$M\^Mss_U\/\]\&LK9\[5vQub\^w\[KDD\<EjmhUZ\?\.akp2dF\>qmj\;2\}YWFdYx\.Ap\]hjCPTP\(n28k\+3\;o\&WXqs\/gOXdr\$\:r\'do0\;b4c\(f_Gr\=\"\\4\)\[01T7ajQJvL\$W\~mL_sS\/4h\:x\*\[ZN\=KLs\&L5zX\/\/\>it\,o\:aU\(\;Z\>pW\&T7oP\'2K\^E\:x9\'c\[\%z\-\,64JQ5AeH_G\#KijUKghQw\^\\vea3a\?kka_G\$8\#\`\*kynsxzBLru\'\]k_\[7FrVx\}\^\=\$blx\>s\-N\%j\;D\*aZDnsw\:YKZ\%Q\.Kne9\#hP\?\+b3\(SOvL\,\^\;\&u5\@\?5C5Bhb\=m\-vEh_L15Jl\]U\)0RP6\{q\%L\^_z5E\'Dw6X\b"
-Memory allocation - compiled block : 1764
+Memory allocation - compiled block : 1740
 Memory allocation - code portion   : 1628
 ------------------------------------------------------------------
   0 811 Bra
@@ -182,7 +182,7 @@ Memory allocation - code portion   : 1628
 ------------------------------------------------------------------
 
 /(a(?1)b)/
-Memory allocation - compiled block : 168
+Memory allocation - compiled block : 144
 Memory allocation - code portion   : 32
 ------------------------------------------------------------------
   0  13 Bra
@@ -196,7 +196,7 @@ Memory allocation - code portion   : 32
 ------------------------------------------------------------------
 
 /(a(?1)+b)/
-Memory allocation - compiled block : 176
+Memory allocation - compiled block : 152
 Memory allocation - code portion   : 40
 ------------------------------------------------------------------
   0  17 Bra
@@ -212,7 +212,7 @@ Memory allocation - code portion   : 40
 ------------------------------------------------------------------
 
 /a(?P<name1>b|c)d(?P<longername2>e)/
-Memory allocation - compiled block : 242
+Memory allocation - compiled block : 218
 Memory allocation - code portion   : 54
 ------------------------------------------------------------------
   0  24 Bra
@@ -231,7 +231,7 @@ Memory allocation - code portion   : 54
 ------------------------------------------------------------------
 
 /(?:a(?P<c>c(?P<d>d)))(?P<a>a)/
-Memory allocation - compiled block : 218
+Memory allocation - compiled block : 194
 Memory allocation - code portion   : 64
 ------------------------------------------------------------------
   0  29 Bra
@@ -252,7 +252,7 @@ Memory allocation - code portion   : 64
 ------------------------------------------------------------------
 
 /(?P<a>a)...(?P=a)bbb(?P>a)d/
-Memory allocation - compiled block : 196
+Memory allocation - compiled block : 172
 Memory allocation - code portion   : 54
 ------------------------------------------------------------------
   0  24 Bra
@@ -271,7 +271,7 @@ Memory allocation - code portion   : 54
 ------------------------------------------------------------------
 
 /abc(?C255)de(?C)f/
-Memory allocation - compiled block : 186
+Memory allocation - compiled block : 162
 Memory allocation - code portion   : 50
 ------------------------------------------------------------------
   0  22 Bra
@@ -285,7 +285,7 @@ Memory allocation - code portion   : 50
 ------------------------------------------------------------------
 
 /abcde/auto_callout
-Memory allocation - compiled block : 214
+Memory allocation - compiled block : 190
 Memory allocation - code portion   : 78
 ------------------------------------------------------------------
   0  36 Bra
@@ -305,7 +305,7 @@ Memory allocation - code portion   : 78
 ------------------------------------------------------------------
 
 /\x{100}/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -315,7 +315,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /\x{1000}/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -325,7 +325,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /\x{10000}/utf
-Memory allocation - compiled block : 152
+Memory allocation - compiled block : 128
 Memory allocation - code portion   : 16
 ------------------------------------------------------------------
   0   5 Bra
@@ -335,7 +335,7 @@ Memory allocation - code portion   : 16
 ------------------------------------------------------------------
 
 /\x{100000}/utf
-Memory allocation - compiled block : 152
+Memory allocation - compiled block : 128
 Memory allocation - code portion   : 16
 ------------------------------------------------------------------
   0   5 Bra
@@ -345,7 +345,7 @@ Memory allocation - code portion   : 16
 ------------------------------------------------------------------
 
 /\x{10ffff}/utf
-Memory allocation - compiled block : 152
+Memory allocation - compiled block : 128
 Memory allocation - code portion   : 16
 ------------------------------------------------------------------
   0   5 Bra
@@ -358,7 +358,7 @@ Memory allocation - code portion   : 16
 Failed: error 134 at offset 9: character code point value in \x{} or \o{} is too large
 
 /[\x{ff}]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -368,7 +368,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[\x{100}]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -378,7 +378,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /\x80/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -388,7 +388,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /\xff/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -398,7 +398,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /\x{0041}\x{2262}\x{0391}\x{002e}/I,utf
-Memory allocation - compiled block : 162
+Memory allocation - compiled block : 138
 Memory allocation - code portion   : 26
 ------------------------------------------------------------------
   0  10 Bra
@@ -413,7 +413,7 @@ Last code unit = '.'
 Subject length lower bound = 4
 
 /\x{D55c}\x{ad6d}\x{C5B4}/I,utf
-Memory allocation - compiled block : 158
+Memory allocation - compiled block : 134
 Memory allocation - code portion   : 22
 ------------------------------------------------------------------
   0   8 Bra
@@ -428,7 +428,7 @@ Last code unit = \x{c5b4}
 Subject length lower bound = 3
 
 /\x{65e5}\x{672c}\x{8a9e}/I,utf
-Memory allocation - compiled block : 158
+Memory allocation - compiled block : 134
 Memory allocation - code portion   : 22
 ------------------------------------------------------------------
   0   8 Bra
@@ -443,7 +443,7 @@ Last code unit = \x{8a9e}
 Subject length lower bound = 3
 
 /[\x{100}]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -453,7 +453,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[Z\x{100}]/utf
-Memory allocation - compiled block : 190
+Memory allocation - compiled block : 166
 Memory allocation - code portion   : 54
 ------------------------------------------------------------------
   0  24 Bra
@@ -463,7 +463,7 @@ Memory allocation - code portion   : 54
 ------------------------------------------------------------------
 
 /^[\x{100}\E-\Q\E\x{150}]/utf
-Memory allocation - compiled block : 162
+Memory allocation - compiled block : 138
 Memory allocation - code portion   : 26
 ------------------------------------------------------------------
   0  10 Bra
@@ -474,7 +474,7 @@ Memory allocation - code portion   : 26
 ------------------------------------------------------------------
 
 /^[\QĀ\E-\QŐ\E]/utf
-Memory allocation - compiled block : 162
+Memory allocation - compiled block : 138
 Memory allocation - code portion   : 26
 ------------------------------------------------------------------
   0  10 Bra
@@ -488,7 +488,7 @@ Memory allocation - code portion   : 26
 Failed: error 106 at offset 13: missing terminating ] for character class
 
 /[\p{L}]/
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -498,7 +498,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /[\p{^L}]/
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -508,7 +508,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /[\P{L}]/
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -518,7 +518,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /[\P{^L}]/
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -528,7 +528,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /[abc\p{L}\x{0660}]/utf
-Memory allocation - compiled block : 196
+Memory allocation - compiled block : 172
 Memory allocation - code portion   : 60
 ------------------------------------------------------------------
   0  27 Bra
@@ -538,7 +538,7 @@ Memory allocation - code portion   : 60
 ------------------------------------------------------------------
 
 /[\p{Nd}]/utf
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -548,7 +548,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /[\p{Nd}+-]+/utf
-Memory allocation - compiled block : 194
+Memory allocation - compiled block : 170
 Memory allocation - code portion   : 58
 ------------------------------------------------------------------
   0  26 Bra
@@ -558,7 +558,7 @@ Memory allocation - code portion   : 58
 ------------------------------------------------------------------
 
 /A\x{391}\x{10427}\x{ff3a}\x{1fb0}/i,utf
-Memory allocation - compiled block : 168
+Memory allocation - compiled block : 144
 Memory allocation - code portion   : 32
 ------------------------------------------------------------------
   0  13 Bra
@@ -568,7 +568,7 @@ Memory allocation - code portion   : 32
 ------------------------------------------------------------------
 
 /A\x{391}\x{10427}\x{ff3a}\x{1fb0}/utf
-Memory allocation - compiled block : 168
+Memory allocation - compiled block : 144
 Memory allocation - code portion   : 32
 ------------------------------------------------------------------
   0  13 Bra
@@ -578,7 +578,7 @@ Memory allocation - code portion   : 32
 ------------------------------------------------------------------
 
 /[\x{105}-\x{109}]/i,utf
-Memory allocation - compiled block : 160
+Memory allocation - compiled block : 136
 Memory allocation - code portion   : 24
 ------------------------------------------------------------------
   0   9 Bra
@@ -588,7 +588,7 @@ Memory allocation - code portion   : 24
 ------------------------------------------------------------------
 
 /( ( (?(1)0|) )*   )/x
-Memory allocation - compiled block : 188
+Memory allocation - compiled block : 164
 Memory allocation - code portion   : 52
 ------------------------------------------------------------------
   0  23 Bra
@@ -607,7 +607,7 @@ Memory allocation - code portion   : 52
 ------------------------------------------------------------------
 
 /(  (?(1)0|)*   )/x
-Memory allocation - compiled block : 178
+Memory allocation - compiled block : 154
 Memory allocation - code portion   : 42
 ------------------------------------------------------------------
   0  18 Bra
@@ -624,7 +624,7 @@ Memory allocation - code portion   : 42
 ------------------------------------------------------------------
 
 /[a]/
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -634,7 +634,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[a]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -644,7 +644,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[\xaa]/
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -654,7 +654,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[\xaa]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -664,7 +664,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[^a]/
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -674,7 +674,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[^a]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -684,7 +684,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[^\xaa]/
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
@@ -694,7 +694,7 @@ Memory allocation - code portion   : 14
 ------------------------------------------------------------------
 
 /[^\xaa]/utf
-Memory allocation - compiled block : 150
+Memory allocation - compiled block : 126
 Memory allocation - code portion   : 14
 ------------------------------------------------------------------
   0   4 Bra
diff --git i/testdata/testoutput8-32-2 w/testdata/testoutput8-32-2
index d76f3aa..4e5c527 100644
--- i/testdata/testoutput8-32-2
+++ w/testdata/testoutput8-32-2
@@ -10,7 +10,7 @@
 #pattern fullbincode,memory
 
 /((?i)b)/
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -22,7 +22,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /(?s)(.*X|^B)/
-Memory allocation - compiled block : 212
+Memory allocation - compiled block : 188
 Memory allocation - code portion   : 76
 ------------------------------------------------------------------
   0  16 Bra
@@ -38,7 +38,7 @@ Memory allocation - code portion   : 76
 ------------------------------------------------------------------
 
 /(?s:.*X|^B)/
-Memory allocation - compiled block : 208
+Memory allocation - compiled block : 184
 Memory allocation - code portion   : 72
 ------------------------------------------------------------------
   0  15 Bra
@@ -54,7 +54,7 @@ Memory allocation - code portion   : 72
 ------------------------------------------------------------------
 
 /^[[:alnum:]]/
-Memory allocation - compiled block : 196
+Memory allocation - compiled block : 172
 Memory allocation - code portion   : 60
 ------------------------------------------------------------------
   0  12 Bra
@@ -65,7 +65,7 @@ Memory allocation - code portion   : 60
 ------------------------------------------------------------------
 
 /#/Ix
-Memory allocation - compiled block : 156
+Memory allocation - compiled block : 132
 Memory allocation - code portion   : 20
 ------------------------------------------------------------------
   0   2 Bra
@@ -78,7 +78,7 @@ Options: extended
 Subject length lower bound = 0
 
 /a#/Ix
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -92,7 +92,7 @@ First code unit = 'a'
 Subject length lower bound = 1
 
 /x?+/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -102,7 +102,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /x++/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -112,7 +112,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /x{1,3}+/
-Memory allocation - compiled block : 176
+Memory allocation - compiled block : 152
 Memory allocation - code portion   : 40
 ------------------------------------------------------------------
   0   7 Bra
@@ -123,7 +123,7 @@ Memory allocation - code portion   : 40
 ------------------------------------------------------------------
 
 /(x)*+/
-Memory allocation - compiled block : 188
+Memory allocation - compiled block : 164
 Memory allocation - code portion   : 52
 ------------------------------------------------------------------
   0  10 Bra
@@ -136,7 +136,7 @@ Memory allocation - code portion   : 52
 ------------------------------------------------------------------
 
 /^((a+)(?U)([ab]+)(?-U)([bc]+)(\w*))/
-Memory allocation - compiled block : 356
+Memory allocation - compiled block : 332
 Memory allocation - code portion   : 220
 ------------------------------------------------------------------
   0  52 Bra
@@ -160,7 +160,7 @@ Memory allocation - code portion   : 220
 ------------------------------------------------------------------
 
 "8J\$WE\<\.rX\+ix\[d1b\!H\#\?vV0vrK\:ZH1\=2M\>iV\;\?aPhFB\<\*vW\@QW\@sO9\}cfZA\-i\'w\%hKd6gt1UJP\,15_\#QY\$M\^Mss_U\/\]\&LK9\[5vQub\^w\[KDD\<EjmhUZ\?\.akp2dF\>qmj\;2\}YWFdYx\.Ap\]hjCPTP\(n28k\+3\;o\&WXqs\/gOXdr\$\:r\'do0\;b4c\(f_Gr\=\"\\4\)\[01T7ajQJvL\$W\~mL_sS\/4h\:x\*\[ZN\=KLs\&L5zX\/\/\>it\,o\:aU\(\;Z\>pW\&T7oP\'2K\^E\:x9\'c\[\%z\-\,64JQ5AeH_G\#KijUKghQw\^\\vea3a\?kka_G\$8\#\`\*kynsxzBLru\'\]k_\[7FrVx\}\^\=\$blx\>s\-N\%j\;D\*aZDnsw\:YKZ\%Q\.Kne9\#hP\?\+b3\(SOvL\,\^\;\&u5\@\?5C5Bhb\=m\-vEh_L15Jl\]U\)0RP6\{q\%L\^_z5E\'Dw6X\b"
-Memory allocation - compiled block : 3432
+Memory allocation - compiled block : 3408
 Memory allocation - code portion   : 3296
 ------------------------------------------------------------------
   0 821 Bra
@@ -171,7 +171,7 @@ Memory allocation - code portion   : 3296
 ------------------------------------------------------------------
 
 "\$\<\.X\+ix\[d1b\!H\#\?vV0vrK\:ZH1\=2M\>iV\;\?aPhFB\<\*vW\@QW\@sO9\}cfZA\-i\'w\%hKd6gt1UJP\,15_\#QY\$M\^Mss_U\/\]\&LK9\[5vQub\^w\[KDD\<EjmhUZ\?\.akp2dF\>qmj\;2\}YWFdYx\.Ap\]hjCPTP\(n28k\+3\;o\&WXqs\/gOXdr\$\:r\'do0\;b4c\(f_Gr\=\"\\4\)\[01T7ajQJvL\$W\~mL_sS\/4h\:x\*\[ZN\=KLs\&L5zX\/\/\>it\,o\:aU\(\;Z\>pW\&T7oP\'2K\^E\:x9\'c\[\%z\-\,64JQ5AeH_G\#KijUKghQw\^\\vea3a\?kka_G\$8\#\`\*kynsxzBLru\'\]k_\[7FrVx\}\^\=\$blx\>s\-N\%j\;D\*aZDnsw\:YKZ\%Q\.Kne9\#hP\?\+b3\(SOvL\,\^\;\&u5\@\?5C5Bhb\=m\-vEh_L15Jl\]U\)0RP6\{q\%L\^_z5E\'Dw6X\b"
-Memory allocation - compiled block : 3392
+Memory allocation - compiled block : 3368
 Memory allocation - code portion   : 3256
 ------------------------------------------------------------------
   0 811 Bra
@@ -182,7 +182,7 @@ Memory allocation - code portion   : 3256
 ------------------------------------------------------------------
 
 /(a(?1)b)/
-Memory allocation - compiled block : 200
+Memory allocation - compiled block : 176
 Memory allocation - code portion   : 64
 ------------------------------------------------------------------
   0  13 Bra
@@ -196,7 +196,7 @@ Memory allocation - code portion   : 64
 ------------------------------------------------------------------
 
 /(a(?1)+b)/
-Memory allocation - compiled block : 216
+Memory allocation - compiled block : 192
 Memory allocation - code portion   : 80
 ------------------------------------------------------------------
   0  17 Bra
@@ -212,7 +212,7 @@ Memory allocation - code portion   : 80
 ------------------------------------------------------------------
 
 /a(?P<name1>b|c)d(?P<longername2>e)/
-Memory allocation - compiled block : 348
+Memory allocation - compiled block : 324
 Memory allocation - code portion   : 108
 ------------------------------------------------------------------
   0  24 Bra
@@ -231,7 +231,7 @@ Memory allocation - code portion   : 108
 ------------------------------------------------------------------
 
 /(?:a(?P<c>c(?P<d>d)))(?P<a>a)/
-Memory allocation - compiled block : 300
+Memory allocation - compiled block : 276
 Memory allocation - code portion   : 128
 ------------------------------------------------------------------
   0  29 Bra
@@ -252,7 +252,7 @@ Memory allocation - code portion   : 128
 ------------------------------------------------------------------
 
 /(?P<a>a)...(?P=a)bbb(?P>a)d/
-Memory allocation - compiled block : 256
+Memory allocation - compiled block : 232
 Memory allocation - code portion   : 108
 ------------------------------------------------------------------
   0  24 Bra
@@ -271,7 +271,7 @@ Memory allocation - code portion   : 108
 ------------------------------------------------------------------
 
 /abc(?C255)de(?C)f/
-Memory allocation - compiled block : 236
+Memory allocation - compiled block : 212
 Memory allocation - code portion   : 100
 ------------------------------------------------------------------
   0  22 Bra
@@ -285,7 +285,7 @@ Memory allocation - code portion   : 100
 ------------------------------------------------------------------
 
 /abcde/auto_callout
-Memory allocation - compiled block : 292
+Memory allocation - compiled block : 268
 Memory allocation - code portion   : 156
 ------------------------------------------------------------------
   0  36 Bra
@@ -305,7 +305,7 @@ Memory allocation - code portion   : 156
 ------------------------------------------------------------------
 
 /\x{100}/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -315,7 +315,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\x{1000}/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -325,7 +325,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\x{10000}/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -335,7 +335,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\x{100000}/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -345,7 +345,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\x{10ffff}/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -358,7 +358,7 @@ Memory allocation - code portion   : 28
 Failed: error 134 at offset 9: character code point value in \x{} or \o{} is too large
 
 /[\x{ff}]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -368,7 +368,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[\x{100}]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -378,7 +378,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\x80/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -388,7 +388,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\xff/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -398,7 +398,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /\x{0041}\x{2262}\x{0391}\x{002e}/I,utf
-Memory allocation - compiled block : 188
+Memory allocation - compiled block : 164
 Memory allocation - code portion   : 52
 ------------------------------------------------------------------
   0  10 Bra
@@ -413,7 +413,7 @@ Last code unit = '.'
 Subject length lower bound = 4
 
 /\x{D55c}\x{ad6d}\x{C5B4}/I,utf
-Memory allocation - compiled block : 180
+Memory allocation - compiled block : 156
 Memory allocation - code portion   : 44
 ------------------------------------------------------------------
   0   8 Bra
@@ -428,7 +428,7 @@ Last code unit = \x{c5b4}
 Subject length lower bound = 3
 
 /\x{65e5}\x{672c}\x{8a9e}/I,utf
-Memory allocation - compiled block : 180
+Memory allocation - compiled block : 156
 Memory allocation - code portion   : 44
 ------------------------------------------------------------------
   0   8 Bra
@@ -443,7 +443,7 @@ Last code unit = \x{8a9e}
 Subject length lower bound = 3
 
 /[\x{100}]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -453,7 +453,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[Z\x{100}]/utf
-Memory allocation - compiled block : 212
+Memory allocation - compiled block : 188
 Memory allocation - code portion   : 76
 ------------------------------------------------------------------
   0  16 Bra
@@ -463,7 +463,7 @@ Memory allocation - code portion   : 76
 ------------------------------------------------------------------
 
 /^[\x{100}\E-\Q\E\x{150}]/utf
-Memory allocation - compiled block : 188
+Memory allocation - compiled block : 164
 Memory allocation - code portion   : 52
 ------------------------------------------------------------------
   0  10 Bra
@@ -474,7 +474,7 @@ Memory allocation - code portion   : 52
 ------------------------------------------------------------------
 
 /^[\QĀ\E-\QŐ\E]/utf
-Memory allocation - compiled block : 188
+Memory allocation - compiled block : 164
 Memory allocation - code portion   : 52
 ------------------------------------------------------------------
   0  10 Bra
@@ -488,7 +488,7 @@ Memory allocation - code portion   : 52
 Failed: error 106 at offset 13: missing terminating ] for character class
 
 /[\p{L}]/
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -498,7 +498,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /[\p{^L}]/
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -508,7 +508,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /[\P{L}]/
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -518,7 +518,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /[\P{^L}]/
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -528,7 +528,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /[abc\p{L}\x{0660}]/utf
-Memory allocation - compiled block : 224
+Memory allocation - compiled block : 200
 Memory allocation - code portion   : 88
 ------------------------------------------------------------------
   0  19 Bra
@@ -538,7 +538,7 @@ Memory allocation - code portion   : 88
 ------------------------------------------------------------------
 
 /[\p{Nd}]/utf
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -548,7 +548,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /[\p{Nd}+-]+/utf
-Memory allocation - compiled block : 220
+Memory allocation - compiled block : 196
 Memory allocation - code portion   : 84
 ------------------------------------------------------------------
   0  18 Bra
@@ -558,7 +558,7 @@ Memory allocation - code portion   : 84
 ------------------------------------------------------------------
 
 /A\x{391}\x{10427}\x{ff3a}\x{1fb0}/i,utf
-Memory allocation - compiled block : 196
+Memory allocation - compiled block : 172
 Memory allocation - code portion   : 60
 ------------------------------------------------------------------
   0  12 Bra
@@ -568,7 +568,7 @@ Memory allocation - code portion   : 60
 ------------------------------------------------------------------
 
 /A\x{391}\x{10427}\x{ff3a}\x{1fb0}/utf
-Memory allocation - compiled block : 196
+Memory allocation - compiled block : 172
 Memory allocation - code portion   : 60
 ------------------------------------------------------------------
   0  12 Bra
@@ -578,7 +578,7 @@ Memory allocation - code portion   : 60
 ------------------------------------------------------------------
 
 /[\x{105}-\x{109}]/i,utf
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0   9 Bra
@@ -588,7 +588,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /( ( (?(1)0|) )*   )/x
-Memory allocation - compiled block : 240
+Memory allocation - compiled block : 216
 Memory allocation - code portion   : 104
 ------------------------------------------------------------------
   0  23 Bra
@@ -607,7 +607,7 @@ Memory allocation - code portion   : 104
 ------------------------------------------------------------------
 
 /(  (?(1)0|)*   )/x
-Memory allocation - compiled block : 220
+Memory allocation - compiled block : 196
 Memory allocation - code portion   : 84
 ------------------------------------------------------------------
   0  18 Bra
@@ -624,7 +624,7 @@ Memory allocation - code portion   : 84
 ------------------------------------------------------------------
 
 /[a]/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -634,7 +634,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[a]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -644,7 +644,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[\xaa]/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -654,7 +654,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[\xaa]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -664,7 +664,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[^a]/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -674,7 +674,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[^a]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -684,7 +684,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[^\xaa]/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
@@ -694,7 +694,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /[^\xaa]/utf
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0   4 Bra
diff --git i/testdata/testoutput8-8-2 w/testdata/testoutput8-8-2
index f3811d9..3896205 100644
--- i/testdata/testoutput8-8-2
+++ w/testdata/testoutput8-8-2
@@ -10,7 +10,7 @@
 #pattern fullbincode,memory
 
 /((?i)b)/
-Memory allocation - compiled block : 153
+Memory allocation - compiled block : 129
 Memory allocation - code portion   : 17
 ------------------------------------------------------------------
   0  13 Bra
@@ -22,7 +22,7 @@ Memory allocation - code portion   : 17
 ------------------------------------------------------------------
 
 /(?s)(.*X|^B)/
-Memory allocation - compiled block : 161
+Memory allocation - compiled block : 137
 Memory allocation - code portion   : 25
 ------------------------------------------------------------------
   0  21 Bra
@@ -38,7 +38,7 @@ Memory allocation - code portion   : 25
 ------------------------------------------------------------------
 
 /(?s:.*X|^B)/
-Memory allocation - compiled block : 159
+Memory allocation - compiled block : 135
 Memory allocation - code portion   : 23
 ------------------------------------------------------------------
   0  19 Bra
@@ -54,7 +54,7 @@ Memory allocation - code portion   : 23
 ------------------------------------------------------------------
 
 /^[[:alnum:]]/
-Memory allocation - compiled block : 177
+Memory allocation - compiled block : 153
 Memory allocation - code portion   : 41
 ------------------------------------------------------------------
   0  37 Bra
@@ -65,7 +65,7 @@ Memory allocation - code portion   : 41
 ------------------------------------------------------------------
 
 /#/Ix
-Memory allocation - compiled block : 143
+Memory allocation - compiled block : 119
 Memory allocation - code portion   : 7
 ------------------------------------------------------------------
   0   3 Bra
@@ -78,7 +78,7 @@ Options: extended
 Subject length lower bound = 0
 
 /a#/Ix
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -92,7 +92,7 @@ First code unit = 'a'
 Subject length lower bound = 1
 
 /x?+/
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -102,7 +102,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /x++/
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -112,7 +112,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /x{1,3}+/
-Memory allocation - compiled block : 149
+Memory allocation - compiled block : 125
 Memory allocation - code portion   : 13
 ------------------------------------------------------------------
   0   9 Bra
@@ -123,7 +123,7 @@ Memory allocation - code portion   : 13
 ------------------------------------------------------------------
 
 /(x)*+/
-Memory allocation - compiled block : 154
+Memory allocation - compiled block : 130
 Memory allocation - code portion   : 18
 ------------------------------------------------------------------
   0  14 Bra
@@ -136,7 +136,7 @@ Memory allocation - code portion   : 18
 ------------------------------------------------------------------
 
 /^((a+)(?U)([ab]+)(?-U)([bc]+)(\w*))/
-Memory allocation - compiled block : 256
+Memory allocation - compiled block : 232
 Memory allocation - code portion   : 120
 ------------------------------------------------------------------
   0 116 Bra
@@ -160,7 +160,7 @@ Memory allocation - code portion   : 120
 ------------------------------------------------------------------
 
 "8J\$WE\<\.rX\+ix\[d1b\!H\#\?vV0vrK\:ZH1\=2M\>iV\;\?aPhFB\<\*vW\@QW\@sO9\}cfZA\-i\'w\%hKd6gt1UJP\,15_\#QY\$M\^Mss_U\/\]\&LK9\[5vQub\^w\[KDD\<EjmhUZ\?\.akp2dF\>qmj\;2\}YWFdYx\.Ap\]hjCPTP\(n28k\+3\;o\&WXqs\/gOXdr\$\:r\'do0\;b4c\(f_Gr\=\"\\4\)\[01T7ajQJvL\$W\~mL_sS\/4h\:x\*\[ZN\=KLs\&L5zX\/\/\>it\,o\:aU\(\;Z\>pW\&T7oP\'2K\^E\:x9\'c\[\%z\-\,64JQ5AeH_G\#KijUKghQw\^\\vea3a\?kka_G\$8\#\`\*kynsxzBLru\'\]k_\[7FrVx\}\^\=\$blx\>s\-N\%j\;D\*aZDnsw\:YKZ\%Q\.Kne9\#hP\?\+b3\(SOvL\,\^\;\&u5\@\?5C5Bhb\=m\-vEh_L15Jl\]U\)0RP6\{q\%L\^_z5E\'Dw6X\b"
-Memory allocation - compiled block : 962
+Memory allocation - compiled block : 938
 Memory allocation - code portion   : 826
 ------------------------------------------------------------------
   0 822 Bra
@@ -171,7 +171,7 @@ Memory allocation - code portion   : 826
 ------------------------------------------------------------------
 
 "\$\<\.X\+ix\[d1b\!H\#\?vV0vrK\:ZH1\=2M\>iV\;\?aPhFB\<\*vW\@QW\@sO9\}cfZA\-i\'w\%hKd6gt1UJP\,15_\#QY\$M\^Mss_U\/\]\&LK9\[5vQub\^w\[KDD\<EjmhUZ\?\.akp2dF\>qmj\;2\}YWFdYx\.Ap\]hjCPTP\(n28k\+3\;o\&WXqs\/gOXdr\$\:r\'do0\;b4c\(f_Gr\=\"\\4\)\[01T7ajQJvL\$W\~mL_sS\/4h\:x\*\[ZN\=KLs\&L5zX\/\/\>it\,o\:aU\(\;Z\>pW\&T7oP\'2K\^E\:x9\'c\[\%z\-\,64JQ5AeH_G\#KijUKghQw\^\\vea3a\?kka_G\$8\#\`\*kynsxzBLru\'\]k_\[7FrVx\}\^\=\$blx\>s\-N\%j\;D\*aZDnsw\:YKZ\%Q\.Kne9\#hP\?\+b3\(SOvL\,\^\;\&u5\@\?5C5Bhb\=m\-vEh_L15Jl\]U\)0RP6\{q\%L\^_z5E\'Dw6X\b"
-Memory allocation - compiled block : 952
+Memory allocation - compiled block : 928
 Memory allocation - code portion   : 816
 ------------------------------------------------------------------
   0 812 Bra
@@ -182,7 +182,7 @@ Memory allocation - code portion   : 816
 ------------------------------------------------------------------
 
 /(a(?1)b)/
-Memory allocation - compiled block : 158
+Memory allocation - compiled block : 134
 Memory allocation - code portion   : 22
 ------------------------------------------------------------------
   0  18 Bra
@@ -196,7 +196,7 @@ Memory allocation - code portion   : 22
 ------------------------------------------------------------------
 
 /(a(?1)+b)/
-Memory allocation - compiled block : 164
+Memory allocation - compiled block : 140
 Memory allocation - code portion   : 28
 ------------------------------------------------------------------
   0  24 Bra
@@ -212,7 +212,7 @@ Memory allocation - code portion   : 28
 ------------------------------------------------------------------
 
 /a(?P<name1>b|c)d(?P<longername2>e)/
-Memory allocation - compiled block : 200
+Memory allocation - compiled block : 176
 Memory allocation - code portion   : 36
 ------------------------------------------------------------------
   0  32 Bra
@@ -231,7 +231,7 @@ Memory allocation - code portion   : 36
 ------------------------------------------------------------------
 
 /(?:a(?P<c>c(?P<d>d)))(?P<a>a)/
-Memory allocation - compiled block : 193
+Memory allocation - compiled block : 169
 Memory allocation - code portion   : 45
 ------------------------------------------------------------------
   0  41 Bra
@@ -252,7 +252,7 @@ Memory allocation - code portion   : 45
 ------------------------------------------------------------------
 
 /(?P<a>a)...(?P=a)bbb(?P>a)d/
-Memory allocation - compiled block : 174
+Memory allocation - compiled block : 150
 Memory allocation - code portion   : 34
 ------------------------------------------------------------------
   0  30 Bra
@@ -271,7 +271,7 @@ Memory allocation - code portion   : 34
 ------------------------------------------------------------------
 
 /abc(?C255)de(?C)f/
-Memory allocation - compiled block : 167
+Memory allocation - compiled block : 143
 Memory allocation - code portion   : 31
 ------------------------------------------------------------------
   0  27 Bra
@@ -285,7 +285,7 @@ Memory allocation - code portion   : 31
 ------------------------------------------------------------------
 
 /abcde/auto_callout
-Memory allocation - compiled block : 189
+Memory allocation - compiled block : 165
 Memory allocation - code portion   : 53
 ------------------------------------------------------------------
   0  49 Bra
@@ -305,7 +305,7 @@ Memory allocation - code portion   : 53
 ------------------------------------------------------------------
 
 /\x{100}/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -315,7 +315,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /\x{1000}/utf
-Memory allocation - compiled block : 147
+Memory allocation - compiled block : 123
 Memory allocation - code portion   : 11
 ------------------------------------------------------------------
   0   7 Bra
@@ -325,7 +325,7 @@ Memory allocation - code portion   : 11
 ------------------------------------------------------------------
 
 /\x{10000}/utf
-Memory allocation - compiled block : 148
+Memory allocation - compiled block : 124
 Memory allocation - code portion   : 12
 ------------------------------------------------------------------
   0   8 Bra
@@ -335,7 +335,7 @@ Memory allocation - code portion   : 12
 ------------------------------------------------------------------
 
 /\x{100000}/utf
-Memory allocation - compiled block : 148
+Memory allocation - compiled block : 124
 Memory allocation - code portion   : 12
 ------------------------------------------------------------------
   0   8 Bra
@@ -345,7 +345,7 @@ Memory allocation - code portion   : 12
 ------------------------------------------------------------------
 
 /\x{10ffff}/utf
-Memory allocation - compiled block : 148
+Memory allocation - compiled block : 124
 Memory allocation - code portion   : 12
 ------------------------------------------------------------------
   0   8 Bra
@@ -358,7 +358,7 @@ Memory allocation - code portion   : 12
 Failed: error 134 at offset 9: character code point value in \x{} or \o{} is too large
 
 /[\x{ff}]/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -368,7 +368,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /[\x{100}]/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -378,7 +378,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /\x80/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -388,7 +388,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /\xff/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -398,7 +398,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /\x{0041}\x{2262}\x{0391}\x{002e}/I,utf
-Memory allocation - compiled block : 154
+Memory allocation - compiled block : 130
 Memory allocation - code portion   : 18
 ------------------------------------------------------------------
   0  14 Bra
@@ -413,7 +413,7 @@ Last code unit = '.'
 Subject length lower bound = 4
 
 /\x{D55c}\x{ad6d}\x{C5B4}/I,utf
-Memory allocation - compiled block : 155
+Memory allocation - compiled block : 131
 Memory allocation - code portion   : 19
 ------------------------------------------------------------------
   0  15 Bra
@@ -428,7 +428,7 @@ Last code unit = \xb4
 Subject length lower bound = 3
 
 /\x{65e5}\x{672c}\x{8a9e}/I,utf
-Memory allocation - compiled block : 155
+Memory allocation - compiled block : 131
 Memory allocation - code portion   : 19
 ------------------------------------------------------------------
   0  15 Bra
@@ -443,7 +443,7 @@ Last code unit = \x9e
 Subject length lower bound = 3
 
 /[\x{100}]/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -453,7 +453,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /[Z\x{100}]/utf
-Memory allocation - compiled block : 183
+Memory allocation - compiled block : 159
 Memory allocation - code portion   : 47
 ------------------------------------------------------------------
   0  43 Bra
@@ -463,7 +463,7 @@ Memory allocation - code portion   : 47
 ------------------------------------------------------------------
 
 /^[\x{100}\E-\Q\E\x{150}]/utf
-Memory allocation - compiled block : 154
+Memory allocation - compiled block : 130
 Memory allocation - code portion   : 18
 ------------------------------------------------------------------
   0  14 Bra
@@ -474,7 +474,7 @@ Memory allocation - code portion   : 18
 ------------------------------------------------------------------
 
 /^[\QĀ\E-\QŐ\E]/utf
-Memory allocation - compiled block : 154
+Memory allocation - compiled block : 130
 Memory allocation - code portion   : 18
 ------------------------------------------------------------------
   0  14 Bra
@@ -488,7 +488,7 @@ Memory allocation - code portion   : 18
 Failed: error 106 at offset 15: missing terminating ] for character class
 
 /[\p{L}]/
-Memory allocation - compiled block : 151
+Memory allocation - compiled block : 127
 Memory allocation - code portion   : 15
 ------------------------------------------------------------------
   0  11 Bra
@@ -498,7 +498,7 @@ Memory allocation - code portion   : 15
 ------------------------------------------------------------------
 
 /[\p{^L}]/
-Memory allocation - compiled block : 151
+Memory allocation - compiled block : 127
 Memory allocation - code portion   : 15
 ------------------------------------------------------------------
   0  11 Bra
@@ -508,7 +508,7 @@ Memory allocation - code portion   : 15
 ------------------------------------------------------------------
 
 /[\P{L}]/
-Memory allocation - compiled block : 151
+Memory allocation - compiled block : 127
 Memory allocation - code portion   : 15
 ------------------------------------------------------------------
   0  11 Bra
@@ -518,7 +518,7 @@ Memory allocation - code portion   : 15
 ------------------------------------------------------------------
 
 /[\P{^L}]/
-Memory allocation - compiled block : 151
+Memory allocation - compiled block : 127
 Memory allocation - code portion   : 15
 ------------------------------------------------------------------
   0  11 Bra
@@ -528,7 +528,7 @@ Memory allocation - code portion   : 15
 ------------------------------------------------------------------
 
 /[abc\p{L}\x{0660}]/utf
-Memory allocation - compiled block : 186
+Memory allocation - compiled block : 162
 Memory allocation - code portion   : 50
 ------------------------------------------------------------------
   0  46 Bra
@@ -538,7 +538,7 @@ Memory allocation - code portion   : 50
 ------------------------------------------------------------------
 
 /[\p{Nd}]/utf
-Memory allocation - compiled block : 151
+Memory allocation - compiled block : 127
 Memory allocation - code portion   : 15
 ------------------------------------------------------------------
   0  11 Bra
@@ -548,7 +548,7 @@ Memory allocation - code portion   : 15
 ------------------------------------------------------------------
 
 /[\p{Nd}+-]+/utf
-Memory allocation - compiled block : 184
+Memory allocation - compiled block : 160
 Memory allocation - code portion   : 48
 ------------------------------------------------------------------
   0  44 Bra
@@ -558,7 +558,7 @@ Memory allocation - code portion   : 48
 ------------------------------------------------------------------
 
 /A\x{391}\x{10427}\x{ff3a}\x{1fb0}/i,utf
-Memory allocation - compiled block : 161
+Memory allocation - compiled block : 137
 Memory allocation - code portion   : 25
 ------------------------------------------------------------------
   0  21 Bra
@@ -568,7 +568,7 @@ Memory allocation - code portion   : 25
 ------------------------------------------------------------------
 
 /A\x{391}\x{10427}\x{ff3a}\x{1fb0}/utf
-Memory allocation - compiled block : 161
+Memory allocation - compiled block : 137
 Memory allocation - code portion   : 25
 ------------------------------------------------------------------
   0  21 Bra
@@ -578,7 +578,7 @@ Memory allocation - code portion   : 25
 ------------------------------------------------------------------
 
 /[\x{105}-\x{109}]/i,utf
-Memory allocation - compiled block : 153
+Memory allocation - compiled block : 129
 Memory allocation - code portion   : 17
 ------------------------------------------------------------------
   0  13 Bra
@@ -588,7 +588,7 @@ Memory allocation - code portion   : 17
 ------------------------------------------------------------------
 
 /( ( (?(1)0|) )*   )/x
-Memory allocation - compiled block : 174
+Memory allocation - compiled block : 150
 Memory allocation - code portion   : 38
 ------------------------------------------------------------------
   0  34 Bra
@@ -607,7 +607,7 @@ Memory allocation - code portion   : 38
 ------------------------------------------------------------------
 
 /(  (?(1)0|)*   )/x
-Memory allocation - compiled block : 166
+Memory allocation - compiled block : 142
 Memory allocation - code portion   : 30
 ------------------------------------------------------------------
   0  26 Bra
@@ -624,7 +624,7 @@ Memory allocation - code portion   : 30
 ------------------------------------------------------------------
 
 /[a]/
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -634,7 +634,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /[a]/utf
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -644,7 +644,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /[\xaa]/
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -654,7 +654,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /[\xaa]/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
@@ -664,7 +664,7 @@ Memory allocation - code portion   : 10
 ------------------------------------------------------------------
 
 /[^a]/
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -674,7 +674,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /[^a]/utf
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -684,7 +684,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /[^\xaa]/
-Memory allocation - compiled block : 145
+Memory allocation - compiled block : 121
 Memory allocation - code portion   : 9
 ------------------------------------------------------------------
   0   5 Bra
@@ -694,7 +694,7 @@ Memory allocation - code portion   : 9
 ------------------------------------------------------------------
 
 /[^\xaa]/utf
-Memory allocation - compiled block : 146
+Memory allocation - compiled block : 122
 Memory allocation - code portion   : 10
 ------------------------------------------------------------------
   0   6 Bra
