From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Wed, 6 Apr 2022 23:01:11 +0000
Subject: [PATCH] Fix build with recent Meson

---
 data/meson.build | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 585e6d3dd0a3..c4b35cc335fd 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -20,21 +20,18 @@ desktop_in = configure_file(input: 'gnome-nettool.desktop.in.in',
                             output: 'gnome-nettool.desktop.in',
                             configuration: desktop_conf)
 
-i18n.merge_file('desktop',
-                type: 'desktop',
+i18n.merge_file(type: 'desktop',
                 input: desktop_in,
                 output: 'gnome-nettool.desktop',
                 po_dir: po_dir,
                 install: true,
                 install_dir: join_paths(prefix, datadir, 'applications'))
 
-i18n.merge_file('appdata',
-                type: 'xml',
+i18n.merge_file(type: 'xml',
                 input: 'gnome-nettool.appdata.xml.in',
                 output: 'gnome-nettool.appdata.xml',
                 po_dir: po_dir,
                 install: true,
-                install_dir: join_paths(prefix, datadir, 'metainfo'),
-)
+                install_dir: join_paths(prefix, datadir, 'metainfo'))
 
 install_data('org.gnome.gnome-nettool.gschema.xml', install_dir: schemas_dir)
