--- a/install/arch-linux/usr/bin/limine-snapper-notify	2026-07-03 10:27:58.875810137 +0200
+++ b/install/arch-linux/usr/bin/limine-snapper-notify	2026-07-03 10:43:37.273490931 +0200
@@ -92,7 +92,7 @@
 		# Store last notification state
 		last_notify_time=$(date +%s)
 
-	done < <(journalctl --output=short-iso -b -p "$LOG_LEVEL" -n0 -f)
+	done < <($watchcmd)
 }
 
 # Main logic
@@ -101,9 +101,18 @@
 	exit 1
 fi
 
-if ! command -v journalctl &>/dev/null; then
-	echo -e "journalctl is not installed." >&2
-	exit 0
+if [ -f /usr/bin/metalog ] ; then
+  logfile="/var/log/everything/current"
+  if [ -r "$logfile" ] ; then
+    watchcmd="tail -f $logfile"
+  else
+    echo "File $logfile cannot be read by this user."
+    echo "When changing this be carefull to not expose sensitive unformation."
+    exit
+  fi
+else
+  "Syslog file must be available as /var/log/everything/current."
+  exit
 fi
 
 load_config
