diff --git a/setup.py b/setup.py
index 7255a45..5136221 100644
--- a/setup.py
+++ b/setup.py
@@ -6,15 +6,6 @@ import codecs
 import os
 import sys
 
-# Ensure setuptools is available
-try:
-    from ez_setup import use_setuptools
-    use_setuptools()
-except ImportError:
-    # Try to use ez_setup, but if not, continue anyway. The import is known
-    # to fail when installing from a tar.gz.
-    print('Could not import ez_setup', file=sys.stderr)
-
 from setuptools import setup
 
 dbx_mod_path = os.path.join(os.path.dirname(__file__), 'dropbox/dropbox_client.py')
