From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Fri, 18 Apr 2025 03:58:06 +0200
Subject: [PATCH] Remove gtk-doc

---
 configure.ac | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7bbcab84b97d..15b00485c406 100644
--- a/configure.ac
+++ b/configure.ac
@@ -163,32 +163,28 @@ GLIB_GENMARSHAL="`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`"
 AC_SUBST(GLIB_GENMARSHAL)
 
 
-AX_REQUIRE_DEFINED([GTK_DOC_CHECK])
-GTK_DOC_CHECK([1.13],[--flavour no-tmpl])
-
 AC_CONFIG_FILES([
 Makefile
 data/dtd/catalog
 data/xslt/db2html.xsl
 data/xslt/info2html.xsl
 data/xslt/links2html.xsl
 data/xslt/mal2html.xsl
 data/xslt/man2html.xsl
 data/xslt/yelp-common.xsl
 po/Makefile.in
 ])
 
 AC_OUTPUT
 
 dnl ==========================================================================
 echo "
 yelp-$VERSION:
 
 	prefix:                   ${prefix}
 	source code location:	  ${srcdir}
 	compiler:		  ${CC}
 
-	Documentation:            ${enable_gtk_doc}
 	Debug enabled:            ${enable_debug}
 	Enable BZ2:               ${enable_bz2}
 	Enable LZMA:              ${enable_lzma}
