--- a/src/modules/wizard/e_wizard.c	2025-07-08 10:02:31.409767578 +0200
+++ b/src/modules/wizard/e_wizard.c	2025-07-08 10:14:40.746539422 +0200
@@ -241,12 +241,7 @@
 EAPI void
 e_wizard_labels_update(void)
 {
-   if (ecore_file_exists("/etc/bodhi/iso"))
-     {
-        edje_object_part_text_set(o_bg, "e.text.label", _("Next"));
-     }
-   else
-      edje_object_part_text_set(o_bg, "e.text.label", _("Please Wait..."));
+   edje_object_part_text_set(o_bg, "e.text.label", _("Next"));
 }
 
 EAPI const char *
