diff --git a/cmake/qt6xdgiconloader-config.cmake.in b/cmake/qt6xdgiconloader-config.cmake.in
index 71fe212..7bd4f59 100644
--- a/cmake/qt6xdgiconloader-config.cmake.in
+++ b/cmake/qt6xdgiconloader-config.cmake.in
@@ -6,6 +6,10 @@ if (NOT TARGET @QTXDGX_ICONLOADER_LIBRARY_NAME@)
     find_dependency(Qt6Gui @QT_MINIMUM_REQUIRED@)
     find_dependency(Qt6Svg @QT_MINIMUM_REQUIRED@)
 
+    if (Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
+        find_dependency(Qt6GuiPrivate @QT_MINIMUM_REQUIRED@)
+    endif()
+
     if (CMAKE_VERSION VERSION_GREATER 2.8.12)
         cmake_policy(SET CMP0024 NEW)
     endif()
