--- a/configurations/labwc/autostart	2024-12-13 22:03:03.942900310 +0100
+++ b/configurations/labwc/autostart	2024-12-13 22:02:44.692937116 +0100
@@ -6,7 +6,7 @@
 swaybg -i /usr/share/lxqt/wallpapers/origami-dark-labwc.png  >/dev/null 2>&1 &
 
 # Faster startup for GTK apps:
-dbus-update-activation-environment --systemd DISPLAY WAYLAND_DISPLAY > /dev/null 2>&1 &
+dbus-update-activation-environment DISPLAY WAYLAND_DISPLAY > /dev/null 2>&1 &
 
 # Configure output directives such as mode, position, scale and transform.
 # Use wlr-randr to get your output names
