diff -ru hugin-2024.0.1.orig/CMakeLists.txt hugin-2024.0.1/CMakeLists.txt
--- hugin-2024.0.1.orig/CMakeLists.txt	2025-10-17 19:41:57.706846127 +0200
+++ hugin-2024.0.1/CMakeLists.txt	2025-10-17 19:42:21.234447547 +0200
@@ -373,7 +373,7 @@
   ENDIF()
 ENDIF(WIN32)
 
-FIND_PACKAGE(Boost 1.47 COMPONENTS filesystem system)
+FIND_PACKAGE(Boost 1.47 COMPONENTS filesystem)
 
 IF (NOT Boost_FOUND)
   IF(WIN32)
