diff --unified --recursive --text --new-file onionshare-2.6.3.orig/cli/pyproject.toml onionshare-2.6.3/cli/pyproject.toml
--- onionshare-2.6.3.orig/cli/pyproject.toml	2025-10-14 20:44:26.777969617 +0200
+++ onionshare-2.6.3/cli/pyproject.toml	2025-10-14 20:44:36.899696845 +0200
@@ -37,7 +37,7 @@
 packaging = ">=24"
 gevent = "^23.9.1"
 wheel = "^0.41.2"
-cffi = "^1.15.1"
+cffi = ">=1.15.1"
 cython = "^3.0.2"
 qrcode = "^7.4.2"
 
