diff -ru kmymoney-5.2.1.orig/doc/reference.docbook kmymoney-5.2.1/doc/reference.docbook
--- kmymoney-5.2.1.orig/doc/reference.docbook	2025-10-07 14:38:53.058263767 +0200
+++ kmymoney-5.2.1/doc/reference.docbook	2025-10-07 14:39:12.424509466 +0200
@@ -873,8 +873,7 @@
         enabled.  More information may be found in the section on configuring the
         <link linkend="details.settings.plugins.kbanking">KBanking plugin</link> as
         well as in the <ulink
-        url="https://www.gnucash.org/docs/v5/C/gnucash-manual/tools-on-line-banking.html#tool-
-        aqb-modules">&gnucash; documentation</ulink>.
+        url="https://www.gnucash.org/docs/v5/C/gnucash-manual/tools-on-line-banking.html#tool-aqb-modules">&gnucash; documentation</ulink>.
       </para>
 
       <para>
