diff --git a/tests/test_subprocesses.py b/tests/test_subprocesses.py
index adf029a..91f024e 100644
--- a/tests/test_subprocesses.py
+++ b/tests/test_subprocesses.py
@@ -277,9 +277,9 @@ async def test_py39_arguments(
             [sys.executable, "-c", "print('hello')"],
             **{argname: argvalue_factory()},
         )
-    except ValueError as exc:
+    except TypeError as exc:
         if (
-            "unexpected kwargs" in str(exc)
+            "unexpected keyword argument" in str(exc)
             and anyio_backend_name == "asyncio"
             and anyio_backend_options["loop_factory"]
             and anyio_backend_options["loop_factory"].__module__ == "uvloop"
