diff --unified --recursive --text --new-file mat2-0.13.4.orig/tests/test_libmat2.py mat2-0.13.4/tests/test_libmat2.py
--- mat2-0.13.4.orig/tests/test_libmat2.py	2024-11-23 13:24:59.831447231 +0100
+++ mat2-0.13.4/tests/test_libmat2.py	2024-11-23 13:25:12.728866275 +0100
@@ -508,7 +508,7 @@
                 'TrackID': 1,
                 'TrackLayer': 0,
                 'TransferCharacteristics': 'BT.709',
-                'VideoFullRangeFlag': 0,
+                'VideoFullRangeFlag': 'Limited',
             },
         },{
             'name': 'wmv',
