--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,9 +24,9 @@ if(buildcache_program)
   set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${buildcache_program}")
 endif()
 
-set(CMAKE_INSTALL_BINDIR bin CACHE PATH "directory for installing binary files")
-set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "directory for installing library files")
-set(CMAKE_INSTALL_INCLUDEDIR include/lib3mf CACHE PATH "directory for installing header files")
+set(CMAKE_INSTALL_BINDIR bin CACHE STRING "directory for installing binary files")
+set(CMAKE_INSTALL_LIBDIR lib CACHE STRING "directory for installing library files")
+set(CMAKE_INSTALL_INCLUDEDIR include/lib3mf CACHE STRING "directory for installing header files")
 
 option(USE_INCLUDED_ZLIB "Use included zlib" ON)
 option(USE_INCLUDED_LIBZIP "Use included libzip" ON)
