--- a/alacritty/src/event.rs	2024-03-07 23:21:16.277654883 +0100
+++ b/alacritty/src/event.rs	2024-03-07 23:26:34.662236903 +0100
@@ -1308,7 +1308,7 @@
                 EventType::Terminal(event) => match event {
                     TerminalEvent::Title(title) => {
                         if !self.ctx.preserve_title && self.ctx.config.window.dynamic_title {
-                            self.ctx.window().set_title(title);
+                            self.ctx.window().set_title(title + " - Alacritty");
                         }
                     },
                     TerminalEvent::ResetTitle => {
