diff --git a/CMakeLists.txt b/CMakeLists.txt
index cc3ef1400..f7c08d8fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -407,7 +407,7 @@ relative_protobuf_generate_cpp(ONNX_PROTO_SRCS
                                onnx/onnx-operators.in.proto
                                onnx/onnx-data.in.proto)
 
-add_library(onnx_proto ${ONNX_PROTO_SRCS})
+add_library(onnx_proto_object OBJECT ${ONNX_PROTO_SRCS})
 
 file(GLOB_RECURSE __tmp_srcs "${ONNX_ROOT}/onnx/*.h" "${ONNX_ROOT}/onnx/*.cc")
 file(GLOB_RECURSE onnx_gtests_src "${ONNX_ROOT}/onnx/test/cpp/*.h"
@@ -419,8 +419,8 @@ list(REMOVE_ITEM __tmp_srcs ${onnx_gtests_src})
 list(APPEND ONNX_SRCS ${__tmp_srcs})
 
 # Hide all symbols we don't need
-set_target_properties(onnx_proto PROPERTIES CXX_VISIBILITY_PRESET hidden)
-set_target_properties(onnx_proto PROPERTIES VISIBILITY_INLINES_HIDDEN 1)
+set_target_properties(onnx_proto_object PROPERTIES CXX_VISIBILITY_PRESET hidden)
+set_target_properties(onnx_proto_object PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
 
 set(LINKED_PROTOBUF_TARGET protobuf::libprotobuf)
 if(ONNX_USE_LITE_PROTO)
@@ -428,19 +428,12 @@ if(ONNX_USE_LITE_PROTO)
     set(LINKED_PROTOBUF_TARGET protobuf::libprotobuf-lite)
   endif()
 endif()
-target_link_libraries(onnx_proto PUBLIC ${LINKED_PROTOBUF_TARGET})
-foreach(ABSL_USED_TARGET IN LISTS protobuf_ABSL_USED_TARGETS)
-  if(TARGET ${ABSL_USED_TARGET})
-    target_link_libraries(onnx_proto PRIVATE ${ABSL_USED_TARGET})
-  endif()
-endforeach()
-add_onnx_global_defines(onnx_proto)
-target_include_directories(onnx_proto PUBLIC
-  $<BUILD_INTERFACE:${ONNX_ROOT}>
+add_onnx_global_defines(onnx_proto_object)
+target_include_directories(onnx_proto_object PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 if(MSVC)
   # For disabling Protobuf related warnings
-  target_compile_options(onnx_proto PUBLIC
+   set(protobuf_warnings
     /wd4146 # unary minus operator applied to unsigned type,
             # result still unsigned
     /wd4244 # 'argument': conversion from 'google::
@@ -452,18 +445,77 @@ if(MSVC)
   )
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
-  # whole-archive linker option not available on AIX.
-  # So, create a object library
-  add_library(onnx OBJECT ${ONNX_SRCS})
-else()
-  add_library(onnx ${ONNX_SRCS})
-endif()
-set_target_properties(onnx PROPERTIES CXX_VISIBILITY_PRESET hidden)
-set_target_properties(onnx PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
+add_library(onnx_proto)
+
+
+
+target_link_libraries(onnx_proto PUBLIC $<BUILD_INTERFACE:onnx_proto_object>)
+
+
+
+
+
+# onnx_object and onnx_proto_object are collections of C++ object files.
+
+
+# They are introduced to help onnx_cpp2py_export bypass library boundaries and use ONNX private symbols.
+
+
+add_library(onnx_object OBJECT ${ONNX_SRCS})
+
+
+add_dependencies(onnx_object onnx_proto_object)
+
+
+target_include_directories(onnx_object PUBLIC
+
+
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
+
+
+set_target_properties(onnx_object PROPERTIES CXX_VISIBILITY_PRESET hidden)
+
+
+set_target_properties(onnx_object PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
+
 
+target_include_directories(onnx_object PUBLIC $<BUILD_INTERFACE:${ONNX_ROOT}>)
+
+
+add_onnx_global_defines(onnx_object)
+
+
+target_link_libraries(onnx_proto_object PUBLIC ${LINKED_PROTOBUF_TARGET})
+
+
+target_link_libraries(onnx_object PUBLIC ${LINKED_PROTOBUF_TARGET})
+
+
+foreach(ABSL_USED_TARGET IN LISTS protobuf_ABSL_USED_TARGETS)
+
+
+  if(TARGET ${ABSL_USED_TARGET})
+
+
+    target_link_libraries(onnx_proto_object PUBLIC ${ABSL_USED_TARGET})
+
+
+    target_link_libraries(onnx_object PUBLIC ${ABSL_USED_TARGET})
+
+
+  endif()
+
+
+endforeach()
+
+
+add_library(onnx)
+
+
+target_link_libraries(onnx PUBLIC $<BUILD_INTERFACE:onnx_object>)
 target_link_libraries(onnx PUBLIC onnx_proto)
-add_onnx_global_defines(onnx)
+target_compile_options(onnx_object PUBLIC ${protobuf_warnings})
+target_compile_options(onnx_proto_object PUBLIC ${protobuf_warnings})
 
 if(ONNX_BUILD_PYTHON)
   # find system pybind11
@@ -482,35 +534,25 @@ if(ONNX_BUILD_PYTHON)
 
   pybind11_add_module(onnx_cpp2py_export MODULE "${ONNX_ROOT}/onnx/cpp2py_export.cc")
 
-  if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
-    # whole-archive linker option not available on AIX
-    target_sources(onnx_cpp2py_export PRIVATE $<TARGET_OBJECTS:onnx>)
-  else()
-    target_link_libraries(onnx_cpp2py_export PRIVATE $<LINK_LIBRARY:WHOLE_ARCHIVE,onnx>)
-  endif()
-  # Prevent "undefined symbol: _ZNSt10filesystem7__cxx114path14_M_split_cmptsEv"
-  # (std::filesystem::__cxx11::path::_M_split_cmpts()) on gcc 8
-  if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
-    target_link_libraries(onnx_cpp2py_export PRIVATE "-lstdc++fs")
-  endif()
+  target_link_libraries(onnx_cpp2py_export PRIVATE $<BUILD_INTERFACE:onnx_object> $<BUILD_INTERFACE:onnx_proto_object>)
 endif()
 
 if(MSVC)
-  add_msvc_runtime_flag(onnx_proto)
-  add_msvc_runtime_flag(onnx)
+  add_msvc_runtime_flag(onnx_proto_object)
+  add_msvc_runtime_flag(onnx_object)
   if(TARGET onnx_cpp2py_export)
     add_msvc_runtime_flag(onnx_cpp2py_export)
   endif()
   if(ONNX_WERROR)
-    target_compile_options(onnx PRIVATE "/WX")
+    target_compile_options(onnx_object PRIVATE "/WX")
   endif()
 else()
-  target_compile_options(onnx PRIVATE -Wall -Wextra)
+  target_compile_options(onnx_object PRIVATE -Wall -Wextra)
   if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13)
-    target_compile_options(onnx PRIVATE "-Wno-stringop-overflow")
+    target_compile_options(onnx_object PRIVATE "-Wno-stringop-overflow")
   endif()
   if(ONNX_WERROR)
-    target_compile_options(onnx PRIVATE "-Werror")
+    target_compile_options(onnx_object PRIVATE "-Werror")
   endif()
 endif()
 if(ONNX_USE_ASAN AND NOT MSVC)
@@ -575,9 +617,6 @@ if(ONNX_USE_UNITY_BUILD)
 endif()
 
 if(ONNX_BUILD_TESTS)
-  if(BUILD_SHARED_LIBS)
-    message(FATAL_ERROR "Tests requires static build")
-  endif()
   find_package(GTest)
   if(NOT GTest_FOUND)
     include(googletest)
diff --git a/cmake/summary.cmake b/cmake/summary.cmake
index e698111dd..5ded4c4fc 100644
--- a/cmake/summary.cmake
+++ b/cmake/summary.cmake
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Prints accumulated ONNX configuration summary
-function (onnx_print_configuration_summary)
+function(onnx_print_configuration_summary)
   message(STATUS "")
   message(STATUS "******** Summary ********")
   message(STATUS "  CMake version                     : ${CMAKE_VERSION}")
@@ -27,33 +27,33 @@ function (onnx_print_configuration_summary)
   message(STATUS "  BUILD_SHARED_LIBS                 : ${BUILD_SHARED_LIBS}")
   message(STATUS "")
 
-  get_target_property(tmp onnx COMPILE_OPTIONS)
+  get_target_property(tmp onnx_object COMPILE_OPTIONS)
   message(STATUS "  onnx compile options              : ${tmp}")
-  get_target_property(tmp onnx_proto COMPILE_OPTIONS)
+  get_target_property(tmp onnx_proto_object COMPILE_OPTIONS)
   message(STATUS "  onnx_proto compile options        : ${tmp}")
-  get_target_property(tmp onnx COMPILE_DEFINITIONS)
+  get_target_property(tmp onnx_object COMPILE_DEFINITIONS)
   message(STATUS "  onnx compile definitions          : ${tmp}")
-  get_target_property(tmp onnx_proto COMPILE_DEFINITIONS)
+  get_target_property(tmp onnx_proto_object COMPILE_DEFINITIONS)
   message(STATUS "  onnx_proto compile definitions    : ${tmp}")
-  get_target_property(tmp onnx LINK_OPTIONS)
+  get_target_property(tmp onnx_object LINK_OPTIONS)
   if(tmp)
     message(STATUS "  onnx link options                 : ${tmp}")
   endif()
-  get_target_property(tmp onnx_proto LINK_OPTIONS)
+  get_target_property(tmp onnx_proto_object LINK_OPTIONS)
   if(tmp)
     message(STATUS "  onnx_proto link options           : ${tmp}")
   endif()
-  get_target_property(tmp onnx LINK_LIBRARIES)
+  get_target_property(tmp onnx_object LINK_LIBRARIES)
   if(tmp)
     message(STATUS "  onnx link libraries               : ${tmp}")
   endif()
-  get_target_property(tmp onnx_proto LINK_LIBRARIES)
+  get_target_property(tmp onnx_proto_object LINK_LIBRARIES)
   if(tmp)
     message(STATUS "  onnx_proto link libraries         : ${tmp}")
   endif()
 
   message(STATUS "")
-  message(STATUS "  Protobuf version                   : ${Protobuf_VERSION}")
+  message(STATUS "  Protobuf version                  : ${Protobuf_VERSION}")
   if(EXISTS "${ONNX_PROTOC_EXECUTABLE}")
     message(STATUS "  Protobuf compiler                 : ${ONNX_PROTOC_EXECUTABLE}")
   else()
