--- a/screenlayout/gui.py	2025-07-12 21:51:40.370737804 +0200
+++ b/screenlayout/gui.py	2025-07-12 21:47:35.703855270 +0200
@@ -107,7 +107,6 @@
             <toolitem action="Open" />
             <toolitem action="SaveAs" />
             <toolitem action="SaveToStartup" />
-
         </toolbar>
     </ui>
     """
@@ -123,7 +122,7 @@
             ("New", Gtk.STOCK_NEW, None, None, None, self.do_new),
             ("Open", Gtk.STOCK_OPEN, None, None, None, self.do_open),
             ("SaveAs", Gtk.STOCK_SAVE_AS, None, None, None, self.do_save_as),
-            ("SaveToStartup", Gtk.STOCK_SAVE, 'Moksha', None, 'Save to Moksha Startup', self.do_save_to_start),
+            ("SaveToStartup", Gtk.STOCK_SAVE, 'Save to Startup', None, 'Save to Moksha Startup', self.do_save_to_start),
 
             ("Apply", Gtk.STOCK_APPLY, None, '<Control>Return', None, self.do_apply),
             ("LayoutSettings", Gtk.STOCK_PROPERTIES, None,
