--- a/screenlayout/meta.py	2025-07-12 21:51:45.481943952 +0200
+++ b/screenlayout/meta.py	2025-07-12 21:47:35.703960809 +0200
@@ -26,7 +26,7 @@
 # translators, please translate in the style of "Another XRandR GUI
 # (ein weiteres GUI für XRandR)" so users get both the explanation of
 # the acronym and a localized version.
-PROGRAMDESCRIPTION =  _(u'Another XRandR GUI') + '\n\nMoksha Fork.'
+PROGRAMDESCRIPTION = _(u'Another XRandR GUI') + '\n\nMoksha Fork.'
 COPYRIGHT = u'© chrysn 2008 – 2019, Себастьян Gli ţa Κατινα 2011, Johannes Holmberg 2015, actionless 2019'
 
 # other names of contributors found in the git history. mailmap (see
