--- a/screenlayout/widget.py	2025-07-12 21:51:51.878372236 +0200
+++ b/screenlayout/widget.py	2025-07-12 21:47:35.704084088 +0200
@@ -165,8 +165,8 @@
             apps = apps + user_apps.split(':')
         except AttributeError:
             pass
-        moksha_config = os.getenv('E_HOME_DIR')
-        moksha_startup = os.path.expanduser(moksha_config + '/applications/startup/startupcommands')
+
+        moksha_startup = os.path.expanduser('~/.e/e/applications/startup/startupcommands')
         try:
             os.stat(moksha_startup)
             backup = True
@@ -190,6 +190,8 @@
                         sanitized += line
                     else:
                         if rflag and any(cmd in line for cmd in apps):
+                            if line.rstrip('\r*\n').endswith('\\'):
+                                data = data.rstrip('\r*\n') + ' | \\'
                             sanitized += data + '\n'
                             rflag = False
                 start.writelines(sanitized)
@@ -200,6 +202,8 @@
                 sanitized = ''
                 for line in content:
                     if line != '\n':
+                        if not line.rstrip('\r*\n').endswith('\\'):
+                            line = line.rstrip('\r*\n') + ' | \\ \n'
                         sanitized += line
                 start.writelines(sanitized)
                 if sanitized and sanitized[-1] != '\n':
