--- a/scripts/install.sh
+++ b/scripts/install.sh
@@ -101,14 +101,14 @@ esac
 # --- install ------------------------------------------------------------------
 
 # bear-driver and bear-wrapper
-mkdir -p "$DESTDIR$PREFIX/libexec/bear/bin"
-install -m 755 "$SRCDIR/bear-driver" "$DESTDIR$PREFIX/libexec/bear/bin/bear-driver"
-install -m 755 "$SRCDIR/bear-wrapper" "$DESTDIR$PREFIX/libexec/bear/bin/bear-wrapper"
+mkdir -p "$DESTDIR$PREFIX/lib/bear/bin"
+install -m 755 "$SRCDIR/bear-driver" "$DESTDIR$PREFIX/lib/bear/bin/bear-driver"
+install -m 755 "$SRCDIR/bear-wrapper" "$DESTDIR$PREFIX/lib/bear/bin/bear-wrapper"
 
 # preload library (Unix only)
 if [ "$HAS_PRELOAD" = true ] && [ -f "$SRCDIR/$PRELOAD_NAME" ]; then
-    mkdir -p "$DESTDIR$PREFIX/libexec/bear/$INTERCEPT_LIBDIR"
-    install -m 644 "$SRCDIR/$PRELOAD_NAME" "$DESTDIR$PREFIX/libexec/bear/$INTERCEPT_LIBDIR/$PRELOAD_NAME"
+    mkdir -p "$DESTDIR$PREFIX/lib/bear/$INTERCEPT_LIBDIR"
+    install -m 644 "$SRCDIR/$PRELOAD_NAME" "$DESTDIR$PREFIX/lib/bear/$INTERCEPT_LIBDIR/$PRELOAD_NAME"
 fi
 
 # bear entry script
@@ -117,7 +117,7 @@ tmp_bear_sh="$(mktemp)"
 trap 'rm -f "$tmp_bear_sh"' EXIT
 cat > "$tmp_bear_sh" <<ENTRY_SCRIPT
 #!/bin/sh
-exec $PREFIX/libexec/bear/bin/bear-driver "\$@"
+exec $PREFIX/lib/bear/bin/bear-driver "\$@"
 ENTRY_SCRIPT
 install -m 755 "$tmp_bear_sh" "$DESTDIR$PREFIX/bin/bear"
 rm -f "$tmp_bear_sh"
