diff -ru android-tools-35.0.2.orig/vendor/libbase/hex.cpp android-tools-35.0.2/vendor/libbase/hex.cpp
--- android-tools-35.0.2.orig/vendor/libbase/hex.cpp	2026-05-31 00:12:45.746446393 +0200
+++ android-tools-35.0.2/vendor/libbase/hex.cpp	2026-05-31 00:15:22.627635886 +0200
@@ -17,6 +17,7 @@
 #include "android-base/hex.h"
 
 #include "android-base/logging.h"
+#include <cstdint>
 
 namespace android {
 namespace base {
diff -ru android-tools-35.0.2.orig/vendor/libziparchive/include/ziparchive/zip_writer.h android-tools-35.0.2/vendor/libziparchive/include/ziparchive/zip_writer.h
--- android-tools-35.0.2.orig/vendor/libziparchive/include/ziparchive/zip_writer.h	2026-05-31 00:12:45.681099268 +0200
+++ android-tools-35.0.2/vendor/libziparchive/include/ziparchive/zip_writer.h	2026-05-31 00:13:05.203831176 +0200
@@ -24,6 +24,7 @@
 #include <string>
 #include <string_view>
 #include <vector>
+#include <cstdint>
 
 #include "android-base/macros.h"
 #include "android-base/off64_t.h"
