diff --git a/src/crypto/nettle/cipher.c b/src/crypto/nettle/cipher.c
index 0dd2262..be89c89 100644
--- a/src/crypto/nettle/cipher.c
+++ b/src/crypto/nettle/cipher.c
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 #include <nettle/aes.h>
 #include <nettle/eax.h>
+#include <nettle/version.h>
 
 struct crypto_aes_eax {
 	struct eax_aes128_ctx ctx;
@@ -119,7 +120,11 @@ static bool crypto_cipher_aes_eax_encrypt(struct crypto_cipher* self,
 			(uint8_t*)dst->data + dst->len, src);
 	dst->len += src_len;
 
+#if NETTLE_VERSION_MAJOR > 3
+	nettle_eax_aes128_digest(&self->enc_ctx.aes_eax.ctx, mac);
+#else
 	nettle_eax_aes128_digest(&self->enc_ctx.aes_eax.ctx, 16, mac);
+#endif
 
 	return true;
 }
@@ -131,7 +136,11 @@ static ssize_t crypto_cipher_aes_eax_decrypt(struct crypto_cipher* self,
 	crypto_aes_eax_update_nonce(&self->dec_ctx.aes_eax);
 	nettle_eax_aes128_update(&self->dec_ctx.aes_eax.ctx, ad_len, ad);
 	nettle_eax_aes128_decrypt(&self->dec_ctx.aes_eax.ctx, len, dst, src);
+#if NETTLE_VERSION_MAJOR > 3
+	nettle_eax_aes128_digest(&self->dec_ctx.aes_eax.ctx, mac);
+#else
 	nettle_eax_aes128_digest(&self->dec_ctx.aes_eax.ctx, 16, mac);
+#endif
 	return len;
 }
 
@@ -179,7 +188,11 @@ static bool crypto_cipher_aes256_eax_encrypt(struct crypto_cipher* self,
 			(uint8_t*)dst->data + dst->len, src);
 	dst->len += src_len;
 
+#if NETTLE_VERSION_MAJOR > 3
+	EAX_DIGEST(&self->enc_ctx.aes256_eax.ctx, aes256_encrypt, mac);
+#else
 	EAX_DIGEST(&self->enc_ctx.aes256_eax.ctx, aes256_encrypt, 16, mac);
+#endif
 
 	return true;
 }
@@ -191,7 +204,11 @@ static ssize_t crypto_cipher_aes256_eax_decrypt(struct crypto_cipher* self,
 	crypto_aes256_eax_update_nonce(&self->dec_ctx.aes256_eax);
 	EAX_UPDATE(&self->dec_ctx.aes256_eax.ctx, aes256_encrypt, ad_len, ad);
 	EAX_DECRYPT(&self->dec_ctx.aes256_eax.ctx, aes256_encrypt, len, dst, src);
+#if NETTLE_VERSION_MAJOR > 3
+	EAX_DIGEST(&self->dec_ctx.aes256_eax.ctx, aes256_encrypt, mac);
+#else
 	EAX_DIGEST(&self->dec_ctx.aes256_eax.ctx, aes256_encrypt, 16, mac);
+#endif
 	return len;
 }
 
diff --git a/src/crypto/nettle/hash.c b/src/crypto/nettle/hash.c
index 71318f4..9fbfe5f 100644
--- a/src/crypto/nettle/hash.c
+++ b/src/crypto/nettle/hash.c
@@ -21,7 +21,7 @@
 #include <stdlib.h>
 #include <nettle/md5.h>
 #include <nettle/sha1.h>
-#include <nettle/sha.h>
+#include <nettle/sha2.h>
 
 struct crypto_hash {
 	union {
