diff --git a/setup.py b/setup.py
index 676f0e1d..cb6c21a3 100755
--- a/setup.py
+++ b/setup.py
@@ -1404,8 +1404,6 @@ else:
         elif darwin and platform_release_tuple() < (18,):
             # There are still of problems when building on old macos.
             ret.append('libclang==14.0.6')
-        else:
-            ret.append('libclang')
         if msys2:
             print(f'msys2: pip install of swig does not build; assuming `pacman -S swig`.')
         elif openbsd:
@@ -1415,8 +1413,6 @@ else:
         elif darwin:
             # 2025-10-27: new swig-4.4.0 fails badly at runtime.
             ret.append('swig==4.3.1')
-        else:
-            ret.append('swig')
         return ret
 
 
