diff --unified --recursive --text --new-file elixir-1.18.4.orig/lib/mix/test/mix/rebar_test.exs elixir-1.18.4/lib/mix/test/mix/rebar_test.exs
--- elixir-1.18.4.orig/lib/mix/test/mix/rebar_test.exs	2025-09-28 19:52:53.628094304 +0200
+++ elixir-1.18.4/lib/mix/test/mix/rebar_test.exs	2025-09-28 19:54:18.198817226 +0200
@@ -68,36 +68,6 @@
   @git_rebar_charlist ~c"../../test/fixtures/git_rebar"
   @git_rebar_string "../../test/fixtures/git_rebar"
 
-  describe "deps/1" do
-    defp parse_dep(dep) do
-      [deps: [dep]] |> Mix.Rebar.deps() |> hd()
-    end
-
-    test "parses Rebar dependencies" do
-      assert parse_dep({:git_rebar, ~c"~> 1.0"}) == {:git_rebar, "~> 1.0", override: true}
-
-      assert parse_dep({:git_rebar, ~c"~> 1.0", {:pkg, :rebar_fork}}) ==
-               {:git_rebar, "~> 1.0", override: true, hex: :rebar_fork}
-
-      assert parse_dep({:git_rebar, {:pkg, :rebar_fork}}) ==
-               {:git_rebar, override: true, hex: :rebar_fork}
-
-      assert parse_dep({:git_rebar, ~c"0.1..*", {:git, @git_rebar_charlist, :main}}) ==
-               {:git_rebar, ~r"0.1..*", override: true, git: @git_rebar_string, ref: "main"}
-
-      assert parse_dep({:git_rebar, {:git, @git_rebar_charlist, :main}}) ==
-               {:git_rebar, override: true, git: @git_rebar_string, ref: "main"}
-
-      assert parse_dep({:git_rebar, ~c"0.1..*", {:git, @git_rebar_charlist}, [:raw]}) ==
-               {:git_rebar, ~r"0.1..*", override: true, git: @git_rebar_string, compile: false}
-
-      assert parse_dep({:git_rebar, ~c"", {:git, @git_rebar_charlist, {:ref, ~c"64691eb"}}}) ==
-               {:git_rebar, ~r"", override: true, git: @git_rebar_string, ref: "64691eb"}
-
-      assert parse_dep(:git_rebar) == {:git_rebar, override: true}
-    end
-  end
-
   describe "apply_overrides/3" do
     test "applies overrides" do
       config = [deps: {:git_rebar, ~c"~> 2.0"}]
diff --unified --recursive --text --new-file elixir-1.18.4.orig/lib/mix/test/mix/tasks/profile.tprof_test.exs elixir-1.18.4/lib/mix/test/mix/tasks/profile.tprof_test.exs
--- elixir-1.18.4.orig/lib/mix/test/mix/tasks/profile.tprof_test.exs	2025-09-28 19:52:53.629926652 +0200
+++ elixir-1.18.4/lib/mix/test/mix/tasks/profile.tprof_test.exs	2025-09-28 19:55:19.501364733 +0200
@@ -78,14 +78,6 @@
     end)
   end
 
-  test "sorts based on memory per call", context do
-    in_tmp(context.test, fn ->
-      assert capture_io(fn ->
-               Tprof.run(["--type", "memory", "--sort", "per_call", "-e", @expr])
-             end) =~ ~r/\n:erlang\.integer_to_binary\/1.*\nEnum\.each\/2/s
-    end)
-  end
-
   test "aggregates totals over all processes", context do
     in_tmp(context.test, fn ->
       assert capture_io(fn ->
