From 26eadd1e22083ecc5044a7d5aa2417368b7a717d Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Mon, 25 Jul 2022 11:36:27 -0500
Subject: [PATCH] no systemd

---
 src/dbus.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/dbus.c b/src/dbus.c
index 4d9b17a..f6cd43e 100644
--- a/src/dbus.c
+++ b/src/dbus.c
@@ -11,8 +11,8 @@
 #include <glib.h>
 #include <glib/gstdio.h>
 #include <gio/gio.h>
-#include <systemd/sd-bus.h>
-#include <systemd/sd-event.h>
+#include <elogind/sd-bus.h>
+#include <elogind/sd-event.h>
 
 #include "_features.h"
 #include "util-internal.h"
-- 
2.37.1

diff --git a/Makefile b/Makefile
index a492516..5131e3b 100644
--- a/Makefile
+++ b/Makefile
@@ -43,8 +43,6 @@ TESTS = \
 	tests/ctests/test_netplan_state \
 	$(NULL)
 
-SYSTEMD_GENERATOR_DIR=$(shell pkg-config --variable=systemdsystemgeneratordir systemd)
-SYSTEMD_UNIT_DIR=$(shell pkg-config --variable=systemdsystemunitdir systemd)
 BASH_COMPLETIONS_DIR=$(shell pkg-config --variable=completionsdir bash-completion || echo "/etc/bash_completion.d")
 
 GCOV ?= gcov
@@ -80,7 +78,7 @@ generate: libnetplan.so.$(NETPLAN_SOVER) generate.o
 	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(filter-out $<,$^) -L. -lnetplan `pkg-config --cflags --libs glib-2.0 gio-2.0 yaml-0.1 uuid`
 
 netplan-dbus: libnetplan.so.$(NETPLAN_SOVER) dbus.o
-	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(filter-out $<,$(patsubst %.h,,$^)) -L. -lnetplan `pkg-config --cflags --libs libsystemd glib-2.0 gio-2.0 yaml-0.1 uuid`
+	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(filter-out $<,$(patsubst %.h,,$^)) -L. -lnetplan `pkg-config --cflags --libs libelogind glib-2.0 gio-2.0 yaml-0.1 uuid`
 
 src/_features.h: src/[^_]*.[hc]
 	printf "#include <stddef.h>\nstatic const char *feature_flags[] __attribute__((__unused__)) = {\n" > $@
diff --git a/dbus/meson.build b/dbus/meson.build
index fc34e2c..919eaa2 100644
--- a/dbus/meson.build
+++ b/dbus/meson.build
@@ -13,7 +13,7 @@ executable(
     features_h,
     include_directories: inc,
     link_with: libnetplan,
-    dependencies: [libsystemd, glib, gio, yaml, uuid],
+    dependencies: [glib, gio, yaml, uuid],
     install_dir: join_paths(get_option('libexecdir'), 'netplan'),
     install: true)
 
diff --git a/meson.build b/meson.build
index b0b5b1e..7195ae7 100644
--- a/meson.build
+++ b/meson.build
@@ -13,13 +13,10 @@ glib = dependency('glib-2.0')
 gio  = dependency('gio-2.0')
 yaml = dependency('yaml-0.1')
 uuid = dependency('uuid')
-libsystemd = dependency('libsystemd')
 
 meson_make_symlink = meson.current_source_dir() + '/tools/meson-make-symlink.sh'
 
-systemd = dependency('systemd')
 completions = dependency('bash-completion')
-systemd_generator_dir = systemd.get_variable(pkgconfig: 'systemdsystemgeneratordir')
 bash_completions_dir = completions.get_variable(pkgconfig: 'completionsdir', default_value: '/etc/bash_completion.d')
 
 # Order: Fedora/Mageia/openSUSE || Debian/Ubuntu
