diff --git a/tools/install-reqs.sh b/tools/install-reqs.sh
index f7a08f90..eed490e7 100755
--- a/tools/install-reqs.sh
+++ b/tools/install-reqs.sh
@@ -1,4 +1,4 @@
 #!/bin/bash
 set -euo pipefail
 echo "Install requirements.yml ..."
-ansible-galaxy collection install -r requirements.yml -p examples/playbooks/collections
+#ansible-galaxy collection install -r requirements.yml -p examples/playbooks/collections
