--- a/test/relation_test.rb
+++ b/test/relation_test.rb
@@ -171,7 +171,7 @@ module Sawyer
       rel  = Sawyer::Relation.from_link(nil, :self, hash)
       map << rel
 
-      assert_equal "{:self_url=>\"/users/1\"}", map.inspect.strip
+      assert_equal "#{{ self_url: "/users/1" }}", map.inspect.strip
     end
   end
 end
diff --git a/test/resource_test.rb b/test/resource_test.rb
index d801081..f09b0bb 100644
--- a/test/resource_test.rb
+++ b/test/resource_test.rb
@@ -201,7 +201,7 @@ module Sawyer
 
     def test_inspect
       resource = Resource.new @agent, :a => 1
-      assert_equal "{:a=>1}", resource.inspect.strip
+      assert_equal "#{{ a: 1 }}", resource.inspect.strip
     end
 
     def test_each
