 src/args.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/args.rs b/src/args.rs
index 69affce..25b2fca 100644
--- a/src/args.rs
+++ b/src/args.rs
@@ -7,7 +7,7 @@ pub struct Args {

     /// Repositories
     #[arg(
-        default_value = "core,extra,community,multilib",
+        default_value = "system,world,galaxy,lib32",
         long,
         use_value_delimiter = true
     )]
