diff --git a/ui-gtk/UgtkApp.c b/ui-gtk/UgtkApp.c
index c62bd51..c4a516b 100644
--- a/ui-gtk/UgtkApp.c
+++ b/ui-gtk/UgtkApp.c
@@ -1769,13 +1769,15 @@ void  ugtk_app_add_default_category (UgtkApp* app)
 {
 	UgetNode*     cnode;
 	UgetCommon*   common;
+	const char*   dir;
 	UgetCategory* category;
 	static int    counts = 0;
 
 	cnode = uget_node_new (NULL);
 	common = ug_info_realloc (cnode->info, UgetCommonInfo);
 	common->name = ug_strdup_printf ("%s %d", _("New"), counts++);
-	common->folder = ug_strdup (g_get_home_dir ());
+	dir = g_get_user_special_dir (G_USER_DIRECTORY_DOWNLOAD);
+	common->folder = ug_strdup (dir ? dir : g_get_home_dir ());
 	category = ug_info_realloc (cnode->info, UgetCategoryInfo);
 	*(char**)ug_array_alloc (&category->schemes, 1) = ug_strdup ("ftps");
 	*(char**)ug_array_alloc (&category->schemes, 1) = ug_strdup ("magnet");
