diff --git a/pandas/_libs/src/vendored/numpy/datetime/np_datetime.c b/pandas/_libs/src/vendored/numpy/datetime/np_datetime.c
index 934c54fafb6..fced39c5952 100644
--- a/pandas/_libs/src/vendored/numpy/datetime/np_datetime.c
+++ b/pandas/_libs/src/vendored/numpy/datetime/np_datetime.c
@@ -22,10 +22,10 @@ This file is derived from NumPy 1.7. See NUMPY_LICENSE.txt
 
 #include <Python.h>
 
-#include "pandas/vendored/numpy/datetime/np_datetime.h"
 
 #define NO_IMPORT_ARRAY
 #define PY_ARRAY_UNIQUE_SYMBOL PANDAS_DATETIME_NUMPY
+#include "pandas/vendored/numpy/datetime/np_datetime.h"
 #include <numpy/ndarrayobject.h>
 #include <numpy/npy_common.h>
 
