diff --git a/src/qt/uiaction.cpp b/src/qt/uiaction.cpp
index 4adb7763e0..d18e460f53 100644
--- a/src/qt/uiaction.cpp
+++ b/src/qt/uiaction.cpp
@@ -177,7 +177,7 @@ static bool SimulateKeyboardKey( KeyAction keyAction, Key key, int modifiers )
     // always produces a lowercase of that char! (which must be in uppercase)
     if ( modifiers == wxMOD_SHIFT && key < 256 )
     {
-        const QChar qChar(key);
+        const QChar qChar = char16_t(key);
         if ( qChar.isLetter() )
         {
             widget->windowHandle() != nullptr ?
