From f522bd06d165b4471f9a6fced59ee04c0c0b2e29 Mon Sep 17 00:00:00 2001
From: artoo <artoo@artixlinux.org>
Date: Sun, 17 Aug 2025 16:04:09 +0200
Subject: [PATCH 1/1] btrfs: fix crc32c module loading

---
 src/modules/initcpiocfg/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/initcpiocfg/main.py b/src/modules/initcpiocfg/main.py
index a70d4ffe9..0ad18de1f 100644
--- a/src/modules/initcpiocfg/main.py
+++ b/src/modules/initcpiocfg/main.py
@@ -243,7 +243,7 @@ def find_initcpio_features(partitions, root_mount_point):
         hooks.extend(["filesystems"])
 
     if uses_btrfs:
-        modules.append("crc32c-intel" if cpuinfo().is_intel else "crc32c")
+        modules.append("crc32c")
     else:
         hooks.append("fsck")
 
-- 
2.50.1

