From 22ddf52ec58455cd1ebfa7ff11e9bc11732b6b1e Mon Sep 17 00:00:00 2001
From: artoo <artoo@artixlinux.org>
Date: Sun, 21 Sep 2025 18:20:32 +0200
Subject: [PATCH 1/1] install agetty conf

---
 conf.d/meson.build | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/conf.d/meson.build b/conf.d/meson.build
index 7c2c99a5..e5f78cb4 100644
--- a/conf.d/meson.build
+++ b/conf.d/meson.build
@@ -60,3 +60,11 @@ elif os == 'netbsd'
 endif
 
 install_data(conf_data, install_dir : conf_d_dir)
+
+if get_option('sysvinit') and os == 'linux'
+  foreach x : get_option('agetty')
+    getty_name = 'agetty.' + x
+    install_data('agetty', install_dir: conf_d_dir,
+      rename: getty_name)
+  endforeach
+endif
-- 
2.51.0

