From 3af17f041cebe748cd9d6584ecbe9b823000f4eb Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Wed, 14 Aug 2024 00:03:24 -0500
Subject: [PATCH] remove-systemd

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 302c189..308f581 100644
--- a/meson.build
+++ b/meson.build
@@ -190,7 +190,7 @@ session_tracking = get_option('session_tracking')
 enable_logind = (session_tracking != 'ConsoleKit')
 if enable_logind
   if session_tracking == 'logind'
-    logind_dep = dependency('libsystemd', not_found_message: 'logind support requested but libsystemd library not found')
+    logind_dep = dependency('libelogind', not_found_message: 'logind support requested but libelogind library not found')
   else
     logind_dep = dependency('libelogind', not_found_message: 'elogind support requested but libelogind library not found')
   endif
-- 
2.46.0

