diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29bfde9a..f0fbe76f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,13 +101,7 @@ if(NOT WIN32)
 endif()
 
 # qwindowkit (frameless window)
-add_fetch_content(
-  TARGET QWindowKit
-  URL https://github.com/atraczyk/qwindowkit.git
-  BRANCH fix-multiarch-libdir
-  PATCHES ${QWINDOWKIT_PATCHES}
-  OPTIONS ${QWINDOWKIT_OPTIONS}
-)
+find_package(QWindowKit)
 
 list(APPEND CLIENT_LIBS QWindowKit::Quick)
 
