From 4f0c64e0428d3ec4bdb231892b8b30c647868faf Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Sun, 14 Nov 2021 11:10:28 -0600
Subject: [PATCH] remove systemd dep

---
 data/meson.build | 10 ----------
 meson.build      | 10 ----------
 2 files changed, 20 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 073bd98..c6e45f5 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -25,13 +25,3 @@ if enable_pam
     install_dir: pk_sysconfdir / 'pam.d',
   )
 endif
-
-if enable_logind
-  configure_file(
-    input: 'polkit.service.in',
-    output: '@BASENAME@',
-    configuration: service_conf,
-    install: true,
-    install_dir: systemd_systemdsystemunitdir,
-  )
-endif
diff --git a/meson.build b/meson.build
index 858078d..701b2d8 100644
--- a/meson.build
+++ b/meson.build
@@ -181,13 +181,6 @@ if enable_logind
   func = 'sd_uid_get_display'
   config_h.set10('HAVE_' + func.to_upper(), cc.has_function(func, dependencies: logind_dep))
 
-  # systemd unit / service files
-  systemd_systemdsystemunitdir = get_option('systemdsystemunitdir')
-  if systemd_systemdsystemunitdir == ''
-    systemd_dep = dependency('systemd', not_found_message: 'systemd required but not found, please provide a valid systemd user unit dir or disable it')
-    # FIXME: systemd.pc file does not use variables with relative paths, so `define_variable` cannot be used
-    systemd_systemdsystemunitdir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
-  endif
 endif
 config_h.set('HAVE_LIBSYSTEMD', enable_logind)
 
@@ -357,9 +350,6 @@ output += '        introspection:            ' + enable_introspection.to_string(
 output += '        Distribution/OS:          ' + os_type + '\n'
 output += '        Authentication framework: ' + auth_fw + '\n'
 output += '        Session tracking:         ' + session_tracking + '\n'
-if enable_logind
-  output += '        systemdsystemunitdir:     ' + systemd_systemdsystemunitdir + '\n'
-endif
 output += '        polkitd user:             ' + polkitd_user + ' \n'
 output += '        PAM support:              ' + enable_pam.to_string() + '\n\n'
 if enable_pam
-- 
2.33.1

