diff --git a/data/meson.build b/data/meson.build
index 651d2cf..c4d2f79 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -26,16 +26,6 @@ if enable_pam
   )
 endif
 
-if session_tracking == 'libsystemd-login'
-  configure_file(
-    input: 'polkit.service.in',
-    output: '@BASENAME@',
-    configuration: service_conf,
-    install: true,
-    install_dir: systemd_systemdsystemunitdir,
-  )
-endif
-
 install_data(
   'policyconfig-1.dtd',
   install_dir: pk_datadir / 'polkit-1'
diff --git a/meson.build b/meson.build
index 3b96562..7a6f0a9 100644
--- a/meson.build
+++ b/meson.build
@@ -207,13 +207,6 @@ if enable_logind
   func = 'sd_uid_get_display'
   config_h.set10('HAVE_' + func.to_upper(), cc.has_function(func, dependencies: logind_dep))
 
-  # systemd unit / service files
-  systemd_systemdsystemunitdir = get_option('systemdsystemunitdir')
-  if systemd_systemdsystemunitdir == '' and session_tracking == 'libsystemd-login'
-    systemd_dep = dependency('systemd', not_found_message: 'systemd required but not found, please provide a valid systemd user unit dir or disable it')
-    # FIXME: systemd.pc file does not use variables with relative paths, so `define_variable` cannot be used
-    systemd_systemdsystemunitdir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
-  endif
 endif
 config_h.set('HAVE_LIBSYSTEMD', enable_logind)
 
diff --git a/meson.build b/meson.build
index 7a6f0a9..4abc73e 100644
--- a/meson.build
+++ b/meson.build
@@ -386,9 +386,6 @@ output += '        introspection:            ' + enable_introspection.to_string(
 output += '        Distribution/OS:          ' + os_type + '\n'
 output += '        Authentication framework: ' + auth_fw + '\n'
 output += '        Session tracking:         ' + session_tracking + '\n'
-if enable_logind
-  output += '        systemdsystemunitdir:     ' + systemd_systemdsystemunitdir + '\n'
-endif
 output += '        polkitd user:             ' + polkitd_user + ' \n'
 output += '        Javascript engine:        ' + js_engine + '\n'
 output += '        PAM support:              ' + enable_pam.to_string() + '\n\n'
