From: ihy123 <aladinandreyy@gmail.com>
Date: Sun, 17 Aug 2025 14:53:52 +0300
Subject: ip/ffmpeg: flush swresample buffer when seeking

---
 ip/ffmpeg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ip/ffmpeg.c b/ip/ffmpeg.c
index 42f630e..775e7de 100644
--- a/ip/ffmpeg.c
+++ b/ip/ffmpeg.c
@@ -444,7 +444,7 @@ static int ffmpeg_seek(struct input_plugin_data *ip_data, double offset)
 
 	priv->swr_frame->nb_samples = 0;
 	avcodec_flush_buffers(priv->codec_ctx);
-	/* also flush swresample buffers? */
+	swr_convert(priv->swr, NULL, 0, NULL, 0); /* flush swr buffer */
 	return 0;
 }
 
