--- a/scripts/repo-add.sh.in	2024-12-01 12:14:03.483914309 +0100
+++ b/scripts/repo-add.sh.in	2024-12-01 12:16:25.341273804 +0100
@@ -77,6 +77,7 @@
 packages to remove can be specified on the command line.\n")"
 		printf -- "\n"
 		printf -- "$(gettext "Options:\n")"
+		printf -- "$(gettext "  -R, --remove      remove old package file from disk after updating database\n")"
 	else
 		printf -- "$(gettext "Please move along, there is nothing to see here.\n")"
 		return
@@ -356,6 +357,11 @@
 	while [[ -n $pkgentry ]]; do
 		notfound=0
 
+		if (( RMEXISTING )); then
+			local oldfilename="$(sed -n '/^%FILENAME%$/ {n;p;q;}' "$pkgentry/desc")"
+			local oldfile="$(dirname "$1")/$oldfilename"
+		fi
+
 		msg2 "$(gettext "Removing existing entry '%s'...")" \
 		"${pkgentry##*/}"
 		rm -rf "$pkgentry"
@@ -367,6 +373,11 @@
 		DB_MODIFIED=1
 
 		pkgentry=$(find_pkgentry "$pkgname")
+		if (( RMEXISTING )); then
+			msg2 "$(gettext "Removing old package file '%s'")" "$oldfilename"
+			rm -f ${oldfile} ${oldfile}.sig
+		fi
+
 	done
 	return $notfound
 } # end db_remove_entry
