From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 1 Aug 2025 07:41:36 +0000
Subject: [PATCH] Replace emblem icon removed from Adwaita

---
 data/ui/row.ui | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/ui/row.ui b/data/ui/row.ui
index 4c1a3ab278df..2aaebf268caf 100644
--- a/data/ui/row.ui
+++ b/data/ui/row.ui
@@ -267,7 +267,7 @@
                                 <property name="receives_default">False</property>
                                 <property name="tooltip_text" translatable="yes">Save</property>
                                 <property name="action_name">recording.save</property>
-                                <property name="icon_name">emblem-ok-symbolic</property>
+                                <property name="icon_name">object-select-symbolic</property>
                                 <style>
                                   <class name="circular"/>
                                   <class name="suggested-action"/>
