--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,14 +43,15 @@ set(ZSTD_URL_HASH "SHA256=eb33e51f49a15e023950cd7825ca74a4a2b43db8354825ac24fc1b
 set(CURL_VERSION "curl-8_21_0")
 set(CURL_URL_HASH "SHA256=ec753aa6f408a3ca9f0d6d5f7a77417aecd1544db13c03ae5d443612bf367364")
 
-set(ZLIB_URL "https://github.com/madler/zlib/releases/download/v${ZLIB_VERSION}/zlib-${ZLIB_VERSION}.tar.gz")
-set(ZSTD_URL "https://github.com/facebook/zstd/releases/download/v${ZSTD_VERSION}/zstd-${ZSTD_VERSION}.tar.gz")
-set(BROTLI_URL "https://github.com/google/brotli/archive/refs/tags/v${BROTLI_VERSION}.tar.gz")
-set(BORINGSSL_URL "https://github.com/google/boringssl/archive/${BORINGSSL_COMMIT}.zip")
-set(NGHTTP2_URL "https://github.com/nghttp2/nghttp2/releases/download/v${NGHTTP2_VERSION}/nghttp2-${NGHTTP2_VERSION}.tar.bz2")
-set(NGTCP2_URL "https://github.com/ngtcp2/ngtcp2/releases/download/v${NGTCP2_VERSION}/ngtcp2-${NGTCP2_VERSION}.tar.bz2")
-set(NGHTTP3_URL "https://github.com/ngtcp2/nghttp3/releases/download/v${NGHTTP3_VERSION}/nghttp3-${NGHTTP3_VERSION}.tar.bz2")
-set(CURL_URL "https://github.com/curl/curl/archive/${CURL_VERSION}.tar.gz")
+set(DEPS_DOWNLOADS_DIR "${CMAKE_BINARY_DIR}/deps/downloads")
+set(ZLIB_URL "${DEPS_DOWNLOADS_DIR}/zlib-${ZLIB_VERSION}.tar.gz")
+set(ZSTD_URL "${DEPS_DOWNLOADS_DIR}/zstd-${ZSTD_VERSION}.tar.gz")
+set(BROTLI_URL "${DEPS_DOWNLOADS_DIR}/brotli-${BROTLI_VERSION}.tar.gz")
+set(BORINGSSL_URL "${DEPS_DOWNLOADS_DIR}/boringssl-${BORINGSSL_COMMIT}.zip")
+set(NGHTTP2_URL "${DEPS_DOWNLOADS_DIR}/nghttp2-${NGHTTP2_VERSION}.tar.bz2")
+set(NGTCP2_URL "${DEPS_DOWNLOADS_DIR}/ngtcp2-${NGTCP2_VERSION}.tar.bz2")
+set(NGHTTP3_URL "${DEPS_DOWNLOADS_DIR}/nghttp3-${NGHTTP3_VERSION}.tar.bz2")
+set(CURL_URL "${DEPS_DOWNLOADS_DIR}/${CURL_VERSION}.tar.gz")
 
 set(_disable_boringssl_asm_default OFF)
 if(WIN32)
