--- a/CMakeLists.txt	2024-02-28 21:19:01.294160486 +0100
+++ b/CMakeLists.txt	2024-02-21 16:39:46.000000000 +0100
@@ -73,25 +73,6 @@
     KF 5.240
 )

-if(CMAKE_SYSTEM_NAME MATCHES Linux)
-    find_package(Systemd 249)
-    set_package_properties(Systemd PROPERTIES TYPE REQUIRED PURPOSE "Coredumpd integration.")
-endif()
-
-if(Systemd_FOUND)
-    find_package(PolkitQt6-1) # This is only used when extracting crashes from coredumpd
-    set_package_properties(PolkitQt6-1 PROPERTIES TYPE REQUIRED PURPOSE "Reading kwin_wayland crashes")
-
-    find_package(coredumpctl)
-    set_package_properties(
-        coredumpctl
-        PROPERTIES
-        TYPE RUNTIME
-        DESCRIPTION "coredumpd CLI. part of systemd"
-        PURPOSE "Required for coredumpd support to work at runtime!"
-    )
-endif()
-
 if(MINGW)
     find_package(ZLIB REQUIRED)
     find_library(INTL_LIBRARY NAMES intl)
