From df1c61bf03eb8598b18e9a81e786fe8991cd95af Mon Sep 17 00:00:00 2001
From: Christian Clason <c.clason@uni-graz.at>
Date: Sat, 3 May 2025 10:35:39 +0200
Subject: [PATCH] build(deps): bump tree-sitter-vim to v0.6.0

---
 runtime/queries/vim/highlights.scm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/runtime/queries/vim/highlights.scm b/runtime/queries/vim/highlights.scm
index df7b3cf483..20a044a60d 100644
--- a/runtime/queries/vim/highlights.scm
+++ b/runtime/queries/vim/highlights.scm
@@ -127,6 +127,7 @@
   "eval"
   "sign"
   "abort"
+  "substitute"
 ] @keyword
 
 (map_statement
@@ -313,6 +314,9 @@
 (binary_operation
   "." @operator)
 
+(lua_statement
+  "=" @keyword)
+
 ; Punctuation
 [
   "("
-- 
2.49.0

