--- a/src/tools/wpctl.c	2026-07-14 13:41:10.284079515 +0200
+++ b/src/tools/wpctl.c	2026-07-21 11:00:06.295666790 +0200
@@ -1896,21 +1896,74 @@
 }
 
 static gboolean
-systemctl_user (const gchar *action, const gchar *service)
+initctl_user (gchar *action, gchar *service)
 {
-  const gchar *argv[] = { "systemctl", "--user", action, service, NULL };
+  const char *fn = "/proc/1/comm";
+  FILE *fl;
+
+  fl = fopen(fn, "r");
+  if (fl == NULL) {
+    fprintf (stderr, "Cannot determine init system\n");
+    return FALSE;
+  }
+  char ch;
+  ch=fgetc(fl);
+  fclose(fl);
+
+  gchar *argv[] = { "", NULL, "", "", NULL };
+
+  switch ((char)ch)
+  {
+    case 'i':
+      argv[0] = "rc-service";
+      argv[1] = "--user";
+      argv[2] = service;
+      argv[3] = action;
+      break;
+    case 'd':
+      argv[0] = "dinitctl";
+      argv[1] = "--user";
+      argv[2] = action;
+      argv[3] = service;
+      break;
+    case 'r':
+      argv[0] = "sv";
+      /* argv[1] = NULL; */
+      argv[2] = action;
+      argv[3] = service;
+      break;
+    case 's':
+      /* s6-rc -l /tmp/${USER}/s6-rc -u change udiskie */
+      /* the -l argument needs to be supplied in order to point to the correct database */
+      gchar *command = "s6-rc -l /tmp/";
+      strcat(command, g_getenv ("USER"));
+      strcat(command, "/s6-rc");
+      argv[0] = command;
+      /* argv[1] = NULL; */
+      if (&action == "start")
+         argv[2] = "-u change";
+      else
+         argv[2] = "-d change";
+      argv[3] = service;
+      break;
+    default:
+      printf("Cannot perform action %s for user service %s on unknown init\n", action, service); 
+      return FALSE;
+      break;
+  }
+
   gint exit_status;
   g_autoptr (GError) error = NULL;
 
   if (!g_spawn_sync (NULL, (gchar **) argv, NULL,
           G_SPAWN_SEARCH_PATH | G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL,
           NULL, NULL, NULL, NULL, &exit_status, &error)) {
-    fprintf (stderr, "Warning: could not run systemctl: %s\n", error->message);
+    fprintf (stderr, "Warning: could not run %s: %s\n", argv[0], error->message);
     return FALSE;
   }
   if (!g_spawn_check_exit_status (exit_status, &error)) {
-    fprintf (stderr, "Warning: systemctl --user %s %s failed: %s\n",
-        action, service, error->message);
+    fprintf (stderr, "Warning: %s %s %s %s failed: %s\n",
+        argv[0], argv[1], argv[2], argv[3], error->message);
     return FALSE;
   }
   return TRUE;
@@ -1998,7 +2051,7 @@
 
   /* Stop wireplumber before removing its files */
   if (do_restart)
-    systemctl_user ("stop", "wireplumber");
+    initctl_user ("stop", "wireplumber");
 
   /* Delete all collected paths in order (children before their parent) */
   for (guint i = 0; i < paths->len; i++) {
@@ -2012,12 +2065,15 @@
   }
 
   /* Restart pipewire if its config was removed */
-  if (do_restart && remove_pw_config)
-    systemctl_user ("restart", "pipewire");
+  if (do_restart && remove_pw_config) {
+    /* initctl_user ("restart", "pipewire"); */
+    initctl_user ("stop", "pipewire");
+    initctl_user ("start", "pipewire");
+  }
 
   /* Start wireplumber again */
   if (do_restart)
-    systemctl_user ("start", "wireplumber");
+    initctl_user ("start", "wireplumber");
 }
 
 #define N_ENTRIES 6
