diff --git a/pyproject.toml b/pyproject.toml
index 6e69866..c2843a8 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=45, <=70.0.0", "setuptools-scm[toml]>=6.2, <=8.1.0"]  # pin max versions of build deps and update as needed
+requires = ["setuptools>=45, <=90.0.0", "setuptools-scm[toml]>=6.2, <=10.0.0"]  # pin max versions of build deps and update as needed
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
