--- automake.orig/m4/tar.m4	2026-09-25 21:51:20.852644136 -0400
+++ automake.new/m4/tar.m4	2026-09-25 21:52:11.165914689 -0400
@@ -87,8 +87,8 @@ m4_if([$1], [v7],
       for _am_tar in tar gnutar gtar; do
         AM_RUN_LOG([$_am_tar --version | grep GNU >/dev/null]) && break
       done
-      am__tar="$_am_tar --format=m4_if([$1], [pax], [posix], [$1]) -chf - "'"$$tardir"'
-      am__tar_="$_am_tar --format=m4_if([$1], [pax], [posix], [$1]) -chf - "'"$tardir"'
+      am__tar="$_am_tar --format=m4_if([$1], [pax], [posix], [$1]) --sort=name --mtime=@0 --owner=0 --group=0 --numeric-owner -chf - "'"$$tardir"'
+      am__tar_="$_am_tar --format=m4_if([$1], [pax], [posix], [$1]) --sort=name --mtime=@0 --owner=0 --group=0 --numeric-owner -chf - "'"$tardir"'
       am__untar="$_am_tar -xf -"
       ;;
     plaintar)
