Subject: [PATCH] allow-system-gradle
---
diff --git a/build-tools/build-infra/build.gradle b/build-tools/build-infra/build.gradle
--- a/build-tools/build-infra/build.gradle
+++ b/build-tools/build-infra/build.gradle
@@ -28,7 +28,6 @@
 // Make sure the build environment is consistent.
 apply from: file('../../gradle/conventions.gradle')
-apply from: file('../../gradle/validation/check-environment.gradle')
 
 // Add spotless/ tidy.
