diff -ru i2pd-2.58.0.orig/build/CMakeLists.txt i2pd-2.58.0/build/CMakeLists.txt
--- i2pd-2.58.0.orig/build/CMakeLists.txt	2025-10-17 19:45:54.739486088 +0200
+++ i2pd-2.58.0/build/CMakeLists.txt	2025-10-17 19:46:55.682109069 +0200
@@ -263,7 +263,7 @@
   endif()
 endif()
 
-find_package(Boost REQUIRED COMPONENTS system filesystem program_options)
+find_package(Boost REQUIRED COMPONENTS filesystem program_options)
 if(NOT DEFINED Boost_FOUND)
   message(SEND_ERROR "Boost is not found, or your boost version was below 1.46. Please download Boost!")
 endif()
