diff --git a/ddclient.in b/ddclient.in
index cba36e0..e08ee45 100755
--- a/ddclient.in
+++ b/ddclient.in
@@ -1912,10 +1912,14 @@ sub _read_config {
     # If file is owned by our effective uid, ensure that it has no access for group or others.
     # Otherwise, require that it isn't writable when not owned by us. For example allow it to
     # be owned by root:ddclient with mode 640. Always ensure that it is not accessible to others.
-    my ($dev, $ino, $mode, @statrest) = stat(FD);
+    my ($dev, $ino, $mode, $nlink, $uid, $gid, @statrest) = stat(FD);
+    my $group_name = 'ddclient';
+    my $group_gid = getgrnam($group_name);
     if ($mode & 077 && -o FD) {
-        if (-f FD && (chmod 0600, $file)) {
+        if ((!defined $group_gid || $gid != $group_gid) && -f FD && (chmod 0600, $file)) {
             warning("file must be accessible only by its owner (fixed)");
+        } elsif ($gid == $group_gid) {
+            info("file is accessible to ddclient group")
         } else {
             warning("file must be accessible only by its owner");
         }
