--- a/pytest.ini
+++ b/pytest.ini
@@ -24,9 +24,9 @@
 addopts = --doctest-modules -m "not network and not heavy and not tui" -p no:unraisableexception -n logical --dist loadfile
 testpaths =
     tests
-    src/blosc2/core.py
-    src/blosc2/ndarray.py
-    src/blosc2/schunk.py
+    blosc2/core.py
+    blosc2/ndarray.py
+    blosc2/schunk.py
 
 markers =
     heavy: tests that take long time to complete.
