diff --git a/fc-solve/source/CMakeLists.txt b/fc-solve/source/CMakeLists.txt
index 95a49d13f..f820c81df 100644
--- a/fc-solve/source/CMakeLists.txt
+++ b/fc-solve/source/CMakeLists.txt
@@ -1018,13 +1018,6 @@ ENDFUNCTION()
 
 IF (NOT "${_PYTHON3}" STREQUAL "_PYTHON3-NOTFOUND")
     SET (found_python3 "1")
-    EXECUTE_PROCESS(COMMAND "python3" "-c" "import random2; import sys; sys.exit(0);" RESULT_VARIABLE _random2)
-    IF ("${_random2}" STREQUAL "0")
-        SET (found_python3_random2 "1")
-    ELSE()
-        SET (found_python3_random2)
-        MESSAGE (WARNING "Could not find the python3's random2 module - please install it.")
-    ENDIF ()
     ASSERT_PYTHON3_MODULE_PRESENCE("six" "")
     ASSERT_PYTHON3_MODULE_PRESENCE("pysol_cards" "")
 ENDIF ()
