diff --git a/run_electrum b/run_electrum
index dc997981c..6529ebf95 100755
--- a/run_electrum
+++ b/run_electrum
@@ -74,8 +74,6 @@ def check_imports():
         import aiorpcx
     except ImportError as e:
         sys.exit(f"Error: {str(e)}. Try 'sudo python3 -m pip install <module-name>'")
-    if not ((0, 22, 0) <= aiorpcx._version < (0, 24)):
-        raise RuntimeError(f'aiorpcX version {aiorpcx._version} does not match required: 0.22.0<=ver<0.24')
     # the following imports are for pyinstaller
     from google.protobuf import descriptor
     from google.protobuf import message
