diff --git a/configure b/configure
index 976f62f..eb16358 100755
--- a/configure
+++ b/configure
@@ -187,7 +187,7 @@ for arg ; do
     --libdir=*) libdir=${arg#*=} ;;
     --includedir=*) includedir=${arg#*=} ;;
     --sysconfdir=*) sysconfdir=${arg#*=} ;;
-    --pkgconfigdir=*) pkgconfigdir=${arg#*=} ;;
+    --pkgconfdir=*) pkgconfdir=${arg#*=} ;;
     --with-sysdeps=*) sysdeps=${arg#*=} manualsysdeps=true ;;
     --with-include=*) var=${arg#*=} ; stripdir var ; addincpath="$addincpath -I$var" ; depincpath="${depincpath}${depincpath:+ }-I$var" ;;
     --with-lib=*) var=${arg#*=} ; stripdir var ; addlibspath="$addlibspath -L$var" ; deplibpath="${deplibpath}${deplibpath:+ }-L$var" ; vpaths="$vpaths $var" ;;
