diff --git a/intern/cycles/kernel/CMakeLists.txt b/intern/cycles/kernel/CMakeLists.txt
index 6b270487abb..800000c05cc 100644
--- a/intern/cycles/kernel/CMakeLists.txt
+++ b/intern/cycles/kernel/CMakeLists.txt
@@ -93,13 +93,12 @@ set(SRC_KERNEL_DEVICE_HIPRT_HEADERS
 set(SRC_KERNEL_DEVICE_HIPRT_SDK
   hiprt/impl/Aabb.h
   hiprt/impl/BvhNode.h
-  hiprt/impl/Geometry.h
+  hiprt/impl/Header.h
   hiprt/impl/hiprt_device_impl.h
   hiprt/impl/hiprt_kernels_bitcode.h
   hiprt/impl/Instance.h
   hiprt/impl/QrDecomposition.h
   hiprt/impl/Quaternion.h
-  hiprt/impl/Scene.h
   hiprt/impl/Transform.h
   hiprt/impl/Triangle.h
 )
