From 12e0d8e791aad3bf8ae82703e00156b37eb8d3e6 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Wed, 7 Jun 2023 03:21:36 -0500
Subject: [PATCH] disable systemd

---
 configure | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/configure b/configure
index 63aaf59..8b1cae3 100755
--- a/configure
+++ b/configure
@@ -15524,18 +15524,7 @@ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
         fi
     fi
 
-    # needs systemd
-    if test -z "$systemdsystemunitdir"; then
-        as_fn_error $? "systemd is required for ZRAM support" "$LINENO" 5
-    fi
-
-    have_zram="yes"
-
-    if test "$have_kbd" = "yes" -a "$have_swap" = "yes"; then
-
-$as_echo "#define HAVE_ZRAM 1" >>confdefs.h
-
-    fi
+    have_zram="no"
 
 
 # Check whether --with-zramconfdir was given.
-- 
2.41.0

