From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 1 Sep 2024 06:46:03 +0000
Subject: [PATCH] Set default window icon

---
 src/photos-application.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/photos-application.c b/src/photos-application.c
index 754bd84039e1..6083235df0c8 100644
--- a/src/photos-application.c
+++ b/src/photos-application.c
@@ -2825,6 +2825,7 @@ photos_application_constructed (GObject *object)
 
   app_id = g_application_get_application_id (G_APPLICATION (self));
   g_set_prgname (app_id);
+  gtk_window_set_default_icon_name (app_id);
 }
 
 
