diff --git a/pyproject.toml b/pyproject.toml
index c040eb3..1114459 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -51,8 +51,8 @@ pyink-use-majority-quotes = true
 requires = [
     # We build against the most recent supported NumPy 2.0 release;
     # see https://github.com/numpy/numpy/issues/27265
-    "numpy~=2.0",
-    "setuptools~=75.7.0",
+    "numpy",
+    "setuptools",
 ]
 build-backend = "setuptools.build_meta"
 
