diff --git i/tests/CLI/modules/vs/vs_create_tests.py w/tests/CLI/modules/vs/vs_create_tests.py
index 468bc533..ab91771e 100644
--- i/tests/CLI/modules/vs/vs_create_tests.py
+++ w/tests/CLI/modules/vs/vs_create_tests.py
@@ -385,7 +385,8 @@ def test_create_like(self, confirm_mock):
         result = self.run_command(['vs', 'create',
                                    '--like=123',
                                    '--san',
-                                   '--billing=hourly'])
+                                   '--billing=hourly'],
+                                  stdin='\n\n\n')
 
         self.assert_no_fail(result)
         self.assertIn('"guid": "1a2b3c-1701"', result.output)
@@ -429,7 +430,8 @@ def test_create_like_tags(self, confirm_mock):
         result = self.run_command(['vs', 'create',
                                    '--like=123',
                                    '--san',
-                                   '--billing=hourly'])
+                                   '--billing=hourly'],
+                                  stdin='\n\n\n')
 
         self.assert_no_fail(result)
         self.assertIn('"guid": "1a2b3c-1701"', result.output)
@@ -460,7 +462,8 @@ def test_create_like_image(self, confirm_mock):
         result = self.run_command(['vs', 'create',
                                    '--like=123',
                                    '--san',
-                                   '--billing=hourly'])
+                                   '--billing=hourly'],
+                                  stdin='\n\n\n')
 
         self.assert_no_fail(result)
         self.assertIn('"guid": "1a2b3c-1701"', result.output)
@@ -499,7 +502,8 @@ def test_create_like_flavor(self, confirm_mock):
         }
 
         confirm_mock.return_value = True
-        result = self.run_command(['vs', 'create', '--like=123'])
+        result = self.run_command(['vs', 'create', '--like=123'],
+                                  stdin='\n\n\n')
 
         self.assert_no_fail(result)
         self.assertIn('"guid": "1a2b3c-1701"', result.output)
@@ -540,7 +544,8 @@ def test_create_like_transient(self, confirm_mock):
         }
 
         confirm_mock.return_value = True
-        result = self.run_command(['vs', 'create', '--like=123'])
+        result = self.run_command(['vs', 'create', '--like=123'],
+                                  stdin='\n\n\n')
 
         self.assert_no_fail(result)
         self.assertIn('"guid": "1a2b3c-1701"', result.output)
